<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Initation</title>

<script src="site_libs/header-attrs-2.21/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<script src="site_libs/clipboard-2.0.6/clipboard.min.js"></script>
<link href="site_libs/xaringanExtra-clipboard-0.2.6/xaringanExtra-clipboard.css" rel="stylesheet" />
<script src="site_libs/xaringanExtra-clipboard-0.2.6/xaringanExtra-clipboard.js"></script>
<script>window.xaringanExtraClipboard(null, {"button":"Copy Code","success":"Copied!","error":"Press Ctrl+C to Copy"})</script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; } /* Alert */
code span.an { color: #008000; } /* Annotation */
code span.at { } /* Attribute */
code span.bu { } /* BuiltIn */
code span.cf { color: #0000ff; } /* ControlFlow */
code span.ch { color: #008080; } /* Char */
code span.cn { color: #585cf6; } /* Constant */
code span.co { color: #4c886b; } /* Comment */
code span.cv { color: #008000; } /* CommentVar */
code span.do { color: #008000; } /* Documentation */
code span.dv { color: #0000cd; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cd; } /* Float */
code span.im { } /* Import */
code span.in { color: #008000; } /* Information */
code span.kw { color: #0000ff; } /* Keyword */
code span.op { color: #687687; } /* Operator */
code span.ot { color: #ff4000; } /* Other */
code span.pp { color: #ff4000; } /* Preprocessor */
code span.sc { color: #008080; } /* SpecialChar */
code span.ss { color: #008080; } /* SpecialString */
code span.st { color: #036a07; } /* String */
code span.va { } /* Variable */
code span.vs { color: #008080; } /* VerbatimString */
code span.wa { color: #008000; font-weight: bold; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<link rel="stylesheet" href="styles.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">OCEAN_5098</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="environment.html">Environment</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Manipulation
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="basics.html">Basics</a>
    </li>
    <li>
      <a href="dplyr.html">dplyr</a>
    </li>
    <li>
      <a href="tidyr.html">tidyr</a>
    </li>
  </ul>
</li>
<li>
  <a href="types.html">Types &amp; structures</a>
</li>
<li>
  <a href="graphs.html">Graphics</a>
</li>
<li>
  <a href="maps.html">Mapping</a>
</li>
<li>
  <a href="functions.html">Loop &amp; functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Statistics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="stat_1.html">Initiation</a>
    </li>
    <li>
      <a href="stat_2.html">Linear models</a>
    </li>
    <li>
      <a href="stat_3.html">Multivariates</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Initation</h1>

</div>


<a href="data:text/x-markdown;base64,---
title: "Initation"
---

```{r eval = TRUE,  message=F, include= FALSE, warning=F, purl=F, results="hide"}
knitr::purl('stat_1.Rmd', documentation = 0)
```

```{r echo=FALSE, purl=F}
xfun::embed_file('stat_1.Rmd')
```


```{r echo=FALSE, purl=F}
xfun::embed_file('stat_1.R')
```

```{r echo=FALSE}
xaringanExtra::use_clipboard()
```

```{r echo=FALSE, purl=F}
xfun::embed_file('data/sf_res.csv')
```


-----------------------

**Statistics, the science of collecting, analyzing, presenting, and interpreting data.**

```{r,  eval=T, warning=F, message=F}
library (psych)
library(ggplot2)
library(dplyr)
library(gridExtra)
library(car)
```

Sample survey methods are used to collect data from observational studies, and experimental design methods are used to collect data from experimental studies. The area of descriptive statistics is concerned primarily with methods of presenting and interpreting data using graphs, tables, and numerical summaries. Whenever statisticians use data from a sample—i.e., a subset of the population—to make statements about a population, they are performing statistical inference. Estimation and hypothesis testing are procedures used to make statistical inferences. Probability plays a key role in statistical inference; it is used to provide measures of the quality and precision of the inferences. 


The data set `sf_res.csv` is extracted from Hsu et al. (2020) and raw data are available [here] (https://github.com/NTU-FRELab/stranger-things.git). The data set describe organismal traits in two soft coral species (see original paper for a full description of variable names, methods and analyses.

```{r}
softies<-read.csv('data/sf_res.csv', sep=',', header=T)
str(softies)
softies$sample_no<-NULL
softies$species<-as.factor(softies$species)
softies$depth<-as.factor(softies$depth)
str(softies)
```
# Descriptive

## all-in-one

Descriptive statistics are extracting basic information from a data set. They are usually used to get a quick idea of possible differences, data distribution, or just summarize them. They are easy to extract on a single variable, but it may be a bit more complicated to get the same information on multiple variables at once. If you are familiar with the pivot table functionality in excel, the same can easily accomplished in R such as here using the functions  as `describe` and/or `describeBy`. Let's explore our `softies` data set

+ A quick `summary` of our data set in R (review): 

```{r,  eval=T}
summary(softies)
```
+ A more detailed summary can be obtained using the function `describe` from the package `psych`.

```{r,  eval=T}
psych::describe(softies[,-c(1:2)])
```

+ This is not very informative as we know we have two species. 

```{r,  eval=T}
psych::describe (softies[,-c(1:2)]~species)
psych::describeBy (softies[,-c(1:2)],softies$species)
```

## Counts & proportions

Simple tabulation (count) summary are made using`table`

```{r,  eval=T}
# One variables
prop.table (table(softies$species))

# two variables
table(softies$depth, softies$species)
prop.table (table(softies$depth, softies$species))

# three variables, with nicer formatting after adding a random variables
softies_2<-softies
softies_2$site<-rep(c("A", "B"),10)
ftable(softies_2$depth, softies_2$species, softies_2$site)
```

## `aggregate` & `apply`

We previously applied a diversity of function on a selected vector, created a filter and applied it on subgroups. In `softies`, the ratio represents the amount of chl c / chl a in the samples. It is informing on the use of light. 

```{r, message=T, results=T, echo=F}
ratio.plot=softies[,c('species','ratio')]
s=ggplot(data=ratio.plot, aes(x=species, y=ratio, fill=species))
ratio_main_title = expression(paste("Ratio chlorophylls", italic(" c/a")))
ratio_y_title = expression(paste("ratio", italic(' c/a'))) 
s + geom_violin(trim = FALSE)+
  geom_jitter(shape=16, position=position_jitter(0.1))+
  labs(title= ratio_main_title, x="Species", y=ratio_y_title)+
  theme_bw()+
  theme(legend.position='none')+
  theme(axis.text.x= element_text(face="italic"))+
  theme(plot.title= element_text(hjust="0.5"))+
  scale_y_continuous(limits=c(0.2,0.6))
```

```{r,  eval=T}
 
mean(softies$ratio)
lit <- softies$species == 'Litophyton sp.'
mean(softies$ratio[lit])
```

Descriptive statistics can be calculated on subgroups using `aggregate` and `apply` functions. The difference between the two is that the 2nd argument in `aggregate` must be a list while in `tapply` it can be a list but this is not mandatory. As a consequence, the output of `aggregate` is a data frame while the one of `tapply` is an array (basically, you deal with the list before or after).   

```{r,  eval=T}
aggregate(softies$ratio,list (species=softies$species),FUN="median")
tapply(softies$ratio,softies$species, FUN="median")
```

The wide family of `apply` functions  represents common alternatives to **loops**. You can read more about the wide application of these functions in many online resources.


<p class="comment">
**Practice 7.1** Using the data `iris` explore how the four traits vary among flower species. (1) Use `boxplot` as visual summaries then extract descriptive statistics by species (`describeBy` or others) (2) Count number of observations in each trait and species (it should result in a table of three species by four traits filled with 50 values) (3) Calculate the `median` of each variable by `Species`, then calculate the `mean` by `Species` but for the `Sepal.Length` only.
</p>

```{r, class.source = "fold-hide",  eval=F}
plot1 <-ggplot(iris, aes(x=Species, y=Sepal.Length)) + 
  geom_boxplot()
  
plot2 <-ggplot(iris, aes(x=Species, y=Sepal.Width)) + 
  geom_boxplot() 
  
plot3 <-ggplot(iris, aes(x=Species, y=Petal.Length)) + 
  geom_boxplot() 
  
plot4 <-ggplot(iris, aes(x=Species, y=Petal.Width)) + 
  geom_boxplot() 
  
grid.arrange(plot1, plot2,plot3, plot4, ncol=2)
describeBy (iris, iris$Species)

iris %>% group_by(Species) %>% summarise(across(c(1:4), length))
aggregate(iris[,1:4],by=list(iris$Species), median)
tapply(iris$Sepal.Length , iris$Species, mean)
```

# Hypothesis testing

One of the most basic concepts in statistics is hypothesis testing and something called **The Null Hypothesis**. This class is not about statitics but some basic concepts and understanding are needed. The amazing YouTube Channel [StatQuest](https://www.youtube.com/c/joshstarmer) can help you with that. Josh Starmer is a professor and **a musical artist** at University of North Carolina at Chapel Hill. You have hours of [enjoyment](https://youtube.com/playlist?list=PLblh5JKOoLUK0FLuzwntyYI10UQFUhsY9) to listen music and learn about stats. Today, let's got to the fundamental of statistics and explanation about [The Null Hypothesis]((https://www.youtube.com/watch?v=0oc49DyA3hU&list=PLblh5JKOoLUK0FLuzwntyYI10UQFUhsY9&index=8) is a perfect introduction to the topic.  

The correct formulation of the **hypotheses** for what you want to test is crucial: *H0* stands for the null hypothesis - no difference is detected (does not mean there is no difference), *H1* stands for the alternative hypothesis - the presence of a difference. 

Two broad categories of statistics can be distinguished parametric (assuming normal distribution & homoscedasticity - video 3) and non parametric (no assumption of normality).

Note: In these tests, it is important to understand the concept of degrees of freedom: It refers to **the maximum number of logically independent values**, i.e. values that have the freedom to vary in the data sample. Look [here](https://www.youtube.com/watch?v=N20rl2llHno&ab_channel=zedstatistics) if you are not clear with this term, as it is the foundation you will need later on.

| Simply, to calculate the degrees of freedom for a sample mean, you can use the formula $df=n-1$, where $n$ is the sample size as we just only one number will determine the mean; the others have the freedom to vary.

## Correlations

Let's first examine what is a correlation. 

* _Pearson_ correlation - parametric
  
  + $r$ pearson coefficient
$$
r = \frac{{}\sum_{i=1}^{n} (x_i - \overline{x})(y_i - \overline{y})}
{\sqrt{\sum_{i=1}^{n} (x_i - \overline{x})^2(y_i - \overline{y})^2}}
$$
  + Formula for the $t$ test fro the correlation coefficient:
$$
t = r\sqrt\frac{n-2}{1-r^2}
$$
___________with degrees of freedom equal to $n-2$


It may look scary for some of you but it is actually very easy to compute (see a demo in step-by step demo in excel [here](.\Data\demo_stats.xlsx)).


  + Example:

```{r class.source = "fold-show",  eval=F}
# dataset hypotheses?
x<-students$height
y<-students$shoesize
s<-students[,1:2] # a matrix
# Pearson correlation
# cor(x,y)
# cor(s)
cor.test(x,y)
```

This relationship is usually represented with it confident interval on a scatter plot.

```{r,  eval=F}
ggplot(students, aes(x = height, y = shoesize)) + 
  geom_point() +
  stat_smooth(method = "lm", col = "red")
```


* _Spearman_ correlation - non-parametric

It is based on ranking, and tests the existence of a monotonic relationship.

```{r,  eval=F}
# Spearman correlation (monotonic)
# cor(x,y, method ='spearman')
cor.test(x,y, method ='spearman')
```

**The example below illustrates on the importance of running both types of correlation** 

```{r,  eval=F}
w<-(1:100)
z<-exp(x)
cor.test(w,z,method='pearson') # super low
cor.test(w,z,method='spearman') #super high
```

<p class="comment">
**Practice 7.2** Play with modifying a value from the student data set and check how $t$ and $p$ values are affected by this change. Examine change in the confidence interval.
</p>

## Chi-square test

It is a goodness-of-fit test: observed vs theoretical. 

+ Casino example

<center>
![](Figures/die.png){width=50%}
</center>

```{r,  eval=F}
#Cast 240 times a die. We counted occurrence of 1,2,3,4,5,6
die<-data.frame(obs=c(55,44,35,45,31,30), row.names=c('Cast1','Cast2','Cast3','Cast4','Cast5','Cast6'))
die #Is this die fair? Define H0 and H1.  
```

Assumption: all results are equally probably (H0, uniform distribution).

```{r,  eval=F}
chisq.test(die)
# I am cheating
```

+ Hardy-Weinberg

A common biological application of a Chi-square test is the Hardy-Weinberg equilibrium. Genotypes at the Hardy-Weinberg equilibrium will follow: *p*2 + 2*pq* + *q*2 = 1 at panmixia (random mating, *p* and *q* represent allele frequency). In this case, we want to compare our observation to a theoretical distribution.

Let's assume 2 alleles (*A* and *T*) with observed genotypes in a population: 750 *AA*, *50* AT, 200 *TT* => *f(A)*: 0.775 / *f(T)*= 0.225. Theoretical distribution should follow: 0.60(*p2*), 0.5 (*2pq*), 0.05 (*q2*) with *p2* + *2pq* + *q2* = 1. Is our population at the HW-equilibrium (H0)?

```{r,  eval=F}
obs <- c(750, 50, 200)
exp <- c(0.60, 0.35, 0.05)
chisq.test (x=obs, p=exp)
```

+ Frequencies

Chi-square test can simply be used to compare the distribution of frequencies in two populations: 

```{r,  eval=F}
F <- matrix(nrow=4,ncol=2,data=c(33,14, 8,18,31,25,14,12))
chisq.test(F) # alternative see `fisher.test`
```

## Student t-test

Several version of t-test exist: one sample compared to a known mean, two samples (with variance equal - Student's or with unequal variance - Welsh's), **paired** (compare two dependent (e.g. before-after) samples. See also the step-by step demo in excel [here](.\Data\demo_stats.xlsx))

```{r,  eval=F}
# One sample
t.test (students$height, mu=170)
# Two sample (with equal variances)
t.test (students$height~students$gender, var.equal = TRUE)
# Two sample (with unequal variances, default option when using t.test) 
t.test (students$height~students$gender)
# Two sample paired t.test
t.test (students$height~students$gender, paired=T)
```

In the case of a two samples assuming equal variances, the *t* statistic is obtained as following:

![](Figures/ttest.png)


<p class="comment">
**Practice 7.3** Using `iris`. Does `Sepal.Lenght` differ between species `setosa` and `versicolor`? Does `Sepal.Lenght` differ between `virginica`  and `versicolor`? Make a plot, define your hypotheses, test hypotheses
</p>


```{r, code_folding = 'Show Solution',  eval=F}
set <- iris[iris$Species == "setosa", ]$Sepal.Length
ver <- iris[iris$Species == "versicolor", ]$Sepal.Length
vir <- iris[iris$Species == "virginica", ]$Sepal.Length

setver <- t.test(set, ver, paired = FALSE, alternative = "two.sided", var.equal = FALSE)
vervir <- t.test(ver, vir, paired = FALSE, alternative = "two.sided", var.equal = FALSE)
```

## Mann-Whitney

A non-parametric solution for the comparison of two samples: Mann-Whitney U-test (independent) or Wilcoxon W-test (dependant)

```{r class.source = "fold-show",  eval=F}
# Normality plot & test
students$height[6]<-132
students$height[10]<-310
students$height[8]<-132
students$height[9]<-210
boxplot(height~gender, students)
qqnorm(students$height) 
qqline(students$height) 
shapiro.test(students$height) # data are not normal 
wilcox.test (students$height~students$gender) 
```

## Variances

Variance tests are used to determine if the variances of two (or more) populations are equal. F-test are generally very sensitive to the non-normality of the data as you can imagine from the previous `boxplot`. The **homoscedasticity** (or **homogeneity of variance**) is an important condition in parametric statistics.

```{r, eval=F}
# Test of variance: we test HO: homogeneous, H1:heterogeneous
fligner.test (students$height ~ students$gender)
```

Another example using the `ToothGrowth` data set (`car` package) on multiple groups

```{r, eval=F}
tg<-ToothGrowth
tg$dose<-factor(tg$dose)
boxplot(len~dose*supp, data=tg)
# also work with: boxplot(len ~ interaction (dose,supp), data=tg)
# or: plot(len ~ interaction (dose,supp), data=tg)
bartlett.test(len~interaction (supp,dose),data=ToothGrowth) # sensitivity non-normality +++
leveneTest(len~interaction (supp,dose),data=ToothGrowth) # sensitivity non-normality ++
fligner.test(len~interaction (supp,dose),data=ToothGrowth) # sensitivity non-normality +
```


<p class="alert">
**Practice 7.4** Create your __own__ `my.t.test` function (e.g. you will not use the above `t.test` function but you will code your own function). You will probably need to check at `?pt` in order to extract the p-value. You can look at `getAnywhere("t.test.default")` for the actual source code of the `t.test` function. Please use your customized function to test  the effect  of `treatment` on _length_ at various _days_ in the `rairuoho` dataset. Interpret.
</p>




" download="stat_1.Rmd">Download stat_1.Rmd</a>
<a href="data:text/plain;base64,eGFyaW5nYW5FeHRyYTo6dXNlX2NsaXBib2FyZCgpDQoNCg0KDQpsaWJyYXJ5IChwc3ljaCkNCmxpYnJhcnkoZ2dwbG90MikNCmxpYnJhcnkoZHBseXIpDQpsaWJyYXJ5KGdyaWRFeHRyYSkNCmxpYnJhcnkoY2FyKQ0KDQpzb2Z0aWVzPC1yZWFkLmNzdignZGF0YS9zZl9yZXMuY3N2Jywgc2VwPScsJywgaGVhZGVyPVQpDQpzdHIoc29mdGllcykNCnNvZnRpZXMkc2FtcGxlX25vPC1OVUxMDQpzb2Z0aWVzJHNwZWNpZXM8LWFzLmZhY3Rvcihzb2Z0aWVzJHNwZWNpZXMpDQpzb2Z0aWVzJGRlcHRoPC1hcy5mYWN0b3Ioc29mdGllcyRkZXB0aCkNCnN0cihzb2Z0aWVzKQ0KDQpzdW1tYXJ5KHNvZnRpZXMpDQoNCnBzeWNoOjpkZXNjcmliZShzb2Z0aWVzWywtYygxOjIpXSkNCg0KcHN5Y2g6OmRlc2NyaWJlIChzb2Z0aWVzWywtYygxOjIpXX5zcGVjaWVzKQ0KcHN5Y2g6OmRlc2NyaWJlQnkgKHNvZnRpZXNbLC1jKDE6MildLHNvZnRpZXMkc3BlY2llcykNCg0KIyBPbmUgdmFyaWFibGVzDQpwcm9wLnRhYmxlICh0YWJsZShzb2Z0aWVzJHNwZWNpZXMpKQ0KDQojIHR3byB2YXJpYWJsZXMNCnRhYmxlKHNvZnRpZXMkZGVwdGgsIHNvZnRpZXMkc3BlY2llcykNCnByb3AudGFibGUgKHRhYmxlKHNvZnRpZXMkZGVwdGgsIHNvZnRpZXMkc3BlY2llcykpDQoNCiMgdGhyZWUgdmFyaWFibGVzLCB3aXRoIG5pY2VyIGZvcm1hdHRpbmcgYWZ0ZXIgYWRkaW5nIGEgcmFuZG9tIHZhcmlhYmxlcw0Kc29mdGllc18yPC1zb2Z0aWVzDQpzb2Z0aWVzXzIkc2l0ZTwtcmVwKGMoIkEiLCAiQiIpLDEwKQ0KZnRhYmxlKHNvZnRpZXNfMiRkZXB0aCwgc29mdGllc18yJHNwZWNpZXMsIHNvZnRpZXNfMiRzaXRlKQ0KDQpyYXRpby5wbG90PXNvZnRpZXNbLGMoJ3NwZWNpZXMnLCdyYXRpbycpXQ0Kcz1nZ3Bsb3QoZGF0YT1yYXRpby5wbG90LCBhZXMoeD1zcGVjaWVzLCB5PXJhdGlvLCBmaWxsPXNwZWNpZXMpKQ0KcmF0aW9fbWFpbl90aXRsZSA9IGV4cHJlc3Npb24ocGFzdGUoIlJhdGlvIGNobG9yb3BoeWxscyIsIGl0YWxpYygiIGMvYSIpKSkNCnJhdGlvX3lfdGl0bGUgPSBleHByZXNzaW9uKHBhc3RlKCJyYXRpbyIsIGl0YWxpYygnIGMvYScpKSkgDQpzICsgZ2VvbV92aW9saW4odHJpbSA9IEZBTFNFKSsNCiAgZ2VvbV9qaXR0ZXIoc2hhcGU9MTYsIHBvc2l0aW9uPXBvc2l0aW9uX2ppdHRlcigwLjEpKSsNCiAgbGFicyh0aXRsZT0gcmF0aW9fbWFpbl90aXRsZSwgeD0iU3BlY2llcyIsIHk9cmF0aW9feV90aXRsZSkrDQogIHRoZW1lX2J3KCkrDQogIHRoZW1lKGxlZ2VuZC5wb3NpdGlvbj0nbm9uZScpKw0KICB0aGVtZShheGlzLnRleHQueD0gZWxlbWVudF90ZXh0KGZhY2U9Iml0YWxpYyIpKSsNCiAgdGhlbWUocGxvdC50aXRsZT0gZWxlbWVudF90ZXh0KGhqdXN0PSIwLjUiKSkrDQogIHNjYWxlX3lfY29udGludW91cyhsaW1pdHM9YygwLjIsMC42KSkNCg0KIA0KbWVhbihzb2Z0aWVzJHJhdGlvKQ0KbGl0IDwtIHNvZnRpZXMkc3BlY2llcyA9PSAnTGl0b3BoeXRvbiBzcC4nDQptZWFuKHNvZnRpZXMkcmF0aW9bbGl0XSkNCg0KYWdncmVnYXRlKHNvZnRpZXMkcmF0aW8sbGlzdCAoc3BlY2llcz1zb2Z0aWVzJHNwZWNpZXMpLEZVTj0ibWVkaWFuIikNCnRhcHBseShzb2Z0aWVzJHJhdGlvLHNvZnRpZXMkc3BlY2llcywgRlVOPSJtZWRpYW4iKQ0KDQojIyBwbG90MSA8LWdncGxvdChpcmlzLCBhZXMoeD1TcGVjaWVzLCB5PVNlcGFsLkxlbmd0aCkpICsNCiMjICAgZ2VvbV9ib3hwbG90KCkNCiMjIA0KIyMgcGxvdDIgPC1nZ3Bsb3QoaXJpcywgYWVzKHg9U3BlY2llcywgeT1TZXBhbC5XaWR0aCkpICsNCiMjICAgZ2VvbV9ib3hwbG90KCkNCiMjIA0KIyMgcGxvdDMgPC1nZ3Bsb3QoaXJpcywgYWVzKHg9U3BlY2llcywgeT1QZXRhbC5MZW5ndGgpKSArDQojIyAgIGdlb21fYm94cGxvdCgpDQojIyANCiMjIHBsb3Q0IDwtZ2dwbG90KGlyaXMsIGFlcyh4PVNwZWNpZXMsIHk9UGV0YWwuV2lkdGgpKSArDQojIyAgIGdlb21fYm94cGxvdCgpDQojIyANCiMjIGdyaWQuYXJyYW5nZShwbG90MSwgcGxvdDIscGxvdDMsIHBsb3Q0LCBuY29sPTIpDQojIyBkZXNjcmliZUJ5IChpcmlzLCBpcmlzJFNwZWNpZXMpDQojIyANCiMjIGlyaXMgJT4lIGdyb3VwX2J5KFNwZWNpZXMpICU+JSBzdW1tYXJpc2UoYWNyb3NzKGMoMTo0KSwgbGVuZ3RoKSkNCiMjIGFnZ3JlZ2F0ZShpcmlzWywxOjRdLGJ5PWxpc3QoaXJpcyRTcGVjaWVzKSwgbWVkaWFuKQ0KIyMgdGFwcGx5KGlyaXMkU2VwYWwuTGVuZ3RoICwgaXJpcyRTcGVjaWVzLCBtZWFuKQ0KDQojIyAjIGRhdGFzZXQgaHlwb3RoZXNlcz8NCiMjIHg8LXN0dWRlbnRzJGhlaWdodA0KIyMgeTwtc3R1ZGVudHMkc2hvZXNpemUNCiMjIHM8LXN0dWRlbnRzWywxOjJdICMgYSBtYXRyaXgNCiMjICMgUGVhcnNvbiBjb3JyZWxhdGlvbg0KIyMgIyBjb3IoeCx5KQ0KIyMgIyBjb3IocykNCiMjIGNvci50ZXN0KHgseSkNCg0KIyMgZ2dwbG90KHN0dWRlbnRzLCBhZXMoeCA9IGhlaWdodCwgeSA9IHNob2VzaXplKSkgKw0KIyMgICBnZW9tX3BvaW50KCkgKw0KIyMgICBzdGF0X3Ntb290aChtZXRob2QgPSAibG0iLCBjb2wgPSAicmVkIikNCg0KIyMgIyBTcGVhcm1hbiBjb3JyZWxhdGlvbiAobW9ub3RvbmljKQ0KIyMgIyBjb3IoeCx5LCBtZXRob2QgPSdzcGVhcm1hbicpDQojIyBjb3IudGVzdCh4LHksIG1ldGhvZCA9J3NwZWFybWFuJykNCg0KIyMgdzwtKDE6MTAwKQ0KIyMgejwtZXhwKHgpDQojIyBjb3IudGVzdCh3LHosbWV0aG9kPSdwZWFyc29uJykgIyBzdXBlciBsb3cNCiMjIGNvci50ZXN0KHcseixtZXRob2Q9J3NwZWFybWFuJykgI3N1cGVyIGhpZ2gNCg0KIyMgI0Nhc3QgMjQwIHRpbWVzIGEgZGllLiBXZSBjb3VudGVkIG9jY3VycmVuY2Ugb2YgMSwyLDMsNCw1LDYNCiMjIGRpZTwtZGF0YS5mcmFtZShvYnM9Yyg1NSw0NCwzNSw0NSwzMSwzMCksIHJvdy5uYW1lcz1jKCdDYXN0MScsJ0Nhc3QyJywnQ2FzdDMnLCdDYXN0NCcsJ0Nhc3Q1JywnQ2FzdDYnKSkNCiMjIGRpZSAjSXMgdGhpcyBkaWUgZmFpcj8gRGVmaW5lIEgwIGFuZCBIMS4NCg0KIyMgY2hpc3EudGVzdChkaWUpDQojIyAjIEkgYW0gY2hlYXRpbmcNCg0KIyMgb2JzIDwtIGMoNzUwLCA1MCwgMjAwKQ0KIyMgZXhwIDwtIGMoMC42MCwgMC4zNSwgMC4wNSkNCiMjIGNoaXNxLnRlc3QgKHg9b2JzLCBwPWV4cCkNCg0KIyMgRiA8LSBtYXRyaXgobnJvdz00LG5jb2w9MixkYXRhPWMoMzMsMTQsIDgsMTgsMzEsMjUsMTQsMTIpKQ0KIyMgY2hpc3EudGVzdChGKSAjIGFsdGVybmF0aXZlIHNlZSBgZmlzaGVyLnRlc3RgDQoNCiMjICMgT25lIHNhbXBsZQ0KIyMgdC50ZXN0IChzdHVkZW50cyRoZWlnaHQsIG11PTE3MCkNCiMjICMgVHdvIHNhbXBsZSAod2l0aCBlcXVhbCB2YXJpYW5jZXMpDQojIyB0LnRlc3QgKHN0dWRlbnRzJGhlaWdodH5zdHVkZW50cyRnZW5kZXIsIHZhci5lcXVhbCA9IFRSVUUpDQojIyAjIFR3byBzYW1wbGUgKHdpdGggdW5lcXVhbCB2YXJpYW5jZXMsIGRlZmF1bHQgb3B0aW9uIHdoZW4gdXNpbmcgdC50ZXN0KQ0KIyMgdC50ZXN0IChzdHVkZW50cyRoZWlnaHR+c3R1ZGVudHMkZ2VuZGVyKQ0KIyMgIyBUd28gc2FtcGxlIHBhaXJlZCB0LnRlc3QNCiMjIHQudGVzdCAoc3R1ZGVudHMkaGVpZ2h0fnN0dWRlbnRzJGdlbmRlciwgcGFpcmVkPVQpDQoNCiMjIHNldCA8LSBpcmlzW2lyaXMkU3BlY2llcyA9PSAic2V0b3NhIiwgXSRTZXBhbC5MZW5ndGgNCiMjIHZlciA8LSBpcmlzW2lyaXMkU3BlY2llcyA9PSAidmVyc2ljb2xvciIsIF0kU2VwYWwuTGVuZ3RoDQojIyB2aXIgPC0gaXJpc1tpcmlzJFNwZWNpZXMgPT0gInZpcmdpbmljYSIsIF0kU2VwYWwuTGVuZ3RoDQojIyANCiMjIHNldHZlciA8LSB0LnRlc3Qoc2V0LCB2ZXIsIHBhaXJlZCA9IEZBTFNFLCBhbHRlcm5hdGl2ZSA9ICJ0d28uc2lkZWQiLCB2YXIuZXF1YWwgPSBGQUxTRSkNCiMjIHZlcnZpciA8LSB0LnRlc3QodmVyLCB2aXIsIHBhaXJlZCA9IEZBTFNFLCBhbHRlcm5hdGl2ZSA9ICJ0d28uc2lkZWQiLCB2YXIuZXF1YWwgPSBGQUxTRSkNCg0KIyMgIyBOb3JtYWxpdHkgcGxvdCAmIHRlc3QNCiMjIHN0dWRlbnRzJGhlaWdodFs2XTwtMTMyDQojIyBzdHVkZW50cyRoZWlnaHRbMTBdPC0zMTANCiMjIHN0dWRlbnRzJGhlaWdodFs4XTwtMTMyDQojIyBzdHVkZW50cyRoZWlnaHRbOV08LTIxMA0KIyMgYm94cGxvdChoZWlnaHR+Z2VuZGVyLCBzdHVkZW50cykNCiMjIHFxbm9ybShzdHVkZW50cyRoZWlnaHQpDQojIyBxcWxpbmUoc3R1ZGVudHMkaGVpZ2h0KQ0KIyMgc2hhcGlyby50ZXN0KHN0dWRlbnRzJGhlaWdodCkgIyBkYXRhIGFyZSBub3Qgbm9ybWFsDQojIyB3aWxjb3gudGVzdCAoc3R1ZGVudHMkaGVpZ2h0fnN0dWRlbnRzJGdlbmRlcikNCg0KIyMgIyBUZXN0IG9mIHZhcmlhbmNlOiB3ZSB0ZXN0IEhPOiBob21vZ2VuZW91cywgSDE6aGV0ZXJvZ2VuZW91cw0KIyMgZmxpZ25lci50ZXN0IChzdHVkZW50cyRoZWlnaHQgfiBzdHVkZW50cyRnZW5kZXIpDQoNCiMjIHRnPC1Ub290aEdyb3d0aA0KIyMgdGckZG9zZTwtZmFjdG9yKHRnJGRvc2UpDQojIyBib3hwbG90KGxlbn5kb3NlKnN1cHAsIGRhdGE9dGcpDQojIyAjIGFsc28gd29yayB3aXRoOiBib3hwbG90KGxlbiB+IGludGVyYWN0aW9uIChkb3NlLHN1cHApLCBkYXRhPXRnKQ0KIyMgIyBvcjogcGxvdChsZW4gfiBpbnRlcmFjdGlvbiAoZG9zZSxzdXBwKSwgZGF0YT10ZykNCiMjIGJhcnRsZXR0LnRlc3QobGVufmludGVyYWN0aW9uIChzdXBwLGRvc2UpLGRhdGE9VG9vdGhHcm93dGgpICMgc2Vuc2l0aXZpdHkgbm9uLW5vcm1hbGl0eSArKysNCiMjIGxldmVuZVRlc3QobGVufmludGVyYWN0aW9uIChzdXBwLGRvc2UpLGRhdGE9VG9vdGhHcm93dGgpICMgc2Vuc2l0aXZpdHkgbm9uLW5vcm1hbGl0eSArKw0KIyMgZmxpZ25lci50ZXN0KGxlbn5pbnRlcmFjdGlvbiAoc3VwcCxkb3NlKSxkYXRhPVRvb3RoR3Jvd3RoKSAjIHNlbnNpdGl2aXR5IG5vbi1ub3JtYWxpdHkgKw0K" download="stat_1.R">Download stat_1.R</a>
<a href="data:text/csv;base64,c3BlY2llcyxzYW1wbGVfbm8sZGVwdGgsSDE1TixIMTNDLEExNU4sQTEzQyxILkMuTixBLkMuTixwZXJjX29tLHBlcmNfaW0sY2VsbC5hZmR3LGNobGEuYWZkdyxjaGxiLmFmZHcsY2hsYy5hZmR3LHJhdGlvLHByb3RaLmFmZHcscHJvdEguYWZkdyxwcm90LmFmZHcsbGlwLmNvbnQsbGlwLmFmZHcNClN0ZXJlb25lcGh0aHlhIHNwLixTQzE5MDAwMSwxMCxOQSwtMjMuMjIzLDUuNTg0LC0xOS4wNjMsTkEsNC4xNzk3ODA2NzcsMTcuMDA2ODAyNzIsODIuOTkzMTk3MjgsNTU0MzIwMDAwLDExMTIuMDYsLTIuMzIsNTQwLjEsMC40ODU2NzUyMzMsODY0MjYuMDIxMDIsNTU3NDMuOTMzNzMsMTQyMTY5Ljk1NDgsMC4wMDcyNTE2MjQsMC4xNjU1MjYyMDINClN0ZXJlb25lcGh0aHlhIHNwLixTQzE5MDAwMywxMCw1LjIwMSwtMjIuNDU5LDUuNDI5LC0xOS4zMDUsNi43Mjk3MzMzOTEsNS4wNzk2MDU4NjUsMTguOTk2NDE1NzcsODEuMDAzNTg0MjMsNjM0NzE2OTgxLjEsMTgyMy40NTI4MywtMjQuNDE1MDk0MzQsNzE1LjMzOTYyMjYsMC4zOTIyOTk0OTQsODI1NzQuNDUyODcsNTUwNjcuNjg1MjIsMTM3NjQyLjEzODEsMC4wMTA1NjY4NDUsMC4xNTIxMDY4MzQNClN0ZXJlb25lcGh0aHlhIHNwLixTQzE5MDAxMCwxMCw0Ljk0MywtMjIuMjUsNC45ODQsLTE5LjE3NSw2LjUxNjQ0OTY3MSw0LjYzMzM1MzA5NCwxOS4yNjYwNTUwNSw4MC43MzM5NDQ5NSwzOTM2NTA3OTMuNywxMDYyLjk2ODI1NCwtMC41NTU1NTU1NTYsNDgyLjcxNDI4NTcsMC40NTQxMTkxOTMsNjA4MjEuOTI2ODYsMzgzNTAuODM4MDYsOTkxNzIuNzY0OTIsMC4wMDYzNjQ0MjEsMC4xMjMyNjI2MDQNClN0ZXJlb25lcGh0aHlhIHNwLixTQzE5MDAxMiwxMCw1LjQyOCwtMjEuNiw1LjUyNiwtMTguODcxLDYuMzU4OTM0NjU4LDUuNTI3OTUyNjU2LDIxLjAzNDQ4Mjc2LDc4Ljk2NTUxNzI0LDUyODc4Njg4NS4yLDExNTMuMzExNDc1LC0xLjUyNDU5MDE2NCw1NTguMjYyMjk1MSwwLjQ4NDA1MTYyNiw1MzQxNS4xODkzMiw0OTMxMy40Mzg2MSwxMDI3MjguNjI3OSwwLjAxMTY5NzI1MywwLjIwNzczMjEzOA0KU3RlcmVvbmVwaHRoeWEgc3AuLFNDMTkwMDEzLDEwLDUuMTk0LC0yMS4xOTMsNS4zNzgsLTE5LjA2LDguMTg4ODk4NDI2LDUuMDA2MTM1OTM3LDIwLDgwLDQwODg4ODg4OC45LDg0MS4zMzMzMzMzLC04Ljc3Nzc3Nzc3OCwzODguNTE4NTE4NSwwLjQ2MTc4OTA0Nyw3NDA4Mi43NjIyMSw0NjEyMS4yMzU3OCwxMjAyMDMuOTk4LDAuMDA3MDQ0Njc0LDAuMTY0MDU4NTU2DQpTdGVyZW9uZXBodGh5YSBzcC4sU0MxOTAwMTQsMTAsNS4wNDQsLTIxLjc0Nyw1LjM3NCwtMTguNzM1LDUuOTYwMDgzMTEyLDQuODg2MzIyMTY1LDIyLjU5MjU5MjU5LDc3LjQwNzQwNzQxLDQzNTQwOTgzNi4xLDE0NTMuNTA4MTk3LC0yMy40OTE4MDMyOCw2MjUuNDA5ODM2MSwwLjQzMDI3NjA5OSw2MTA2My4xNTQxNCw1NjIwOS4yOTEzMiwxMTcyNzIuNDQ1NSwwLjAwODMwMDMwNSwwLjE2ODY4MjUyNw0KU3RlcmVvbmVwaHRoeWEgc3AuLFNDMTkwMDE4LDEwLDUuMjg1LC0yMC45OTksNS42NDEsLTE4LjY0NSw1LjczNTM1MzgyMSw0Ljk3Njk0NTE2MSwxOS43MTMyNjE2NSw4MC4yODY3MzgzNSwzMjcyNzI3MjcuMywxMDk0LjQ5MDkwOSwxLjQ3MjcyNzI3Myw0MjAuNDU0NTQ1NSwwLjM4NDE1NTM1Nyw1OTA2OS4wMzcyNSwzNzY5MC4zMjg1Niw5Njc1OS4zNjU4LDAuMDA2MTg5MzE3LDAuMTA0ODMwNDQxDQpTdGVyZW9uZXBodGh5YSBzcC4sU0MxOTAwMTksMTAsNS4wNTEsLTIxLjgyMiw1LjE4NywtMTguMjYzLDkuNDE4OTA2MTQsNS4zMjk2ODY2NDYsMTkuMjg1NzE0MjksODAuNzE0Mjg1NzEsMjU0MjIyMjIyLjIsMTAxMC43NDA3NDEsLTI2LjExMTExMTExLDQ3MS44NTE4NTE5LDAuNDY2ODM3NjY5LDU3Mjc3LjM2OTc3LDM0NzA4LjUwMTQsOTE5ODUuODcxMTcsMC4wMDgxMjY2MzIsMC4xNDUwNTM2Nw0KU3RlcmVvbmVwaHRoeWEgc3AuLFNDMTkwMDIwLDEwLDUuMDQsLTIwLjczMSxOQSwtMTkuMDg5LDguNjkwOTkyMzIxLE5BLDE2LjQ4NzQ1NTIsODMuNTEyNTQ0OCwzNDM2NTIxNzMuOSw5NjEuODkxMzA0MywxLjUsNDU5LjcxNzM5MTMsMC40Nzc5MzA3MDcsOTE4MjUuNjQ2MzcsNDA0MTIuMDcyMjIsMTMyMjM3LjcxODYsMC4wMDYzMzcwMzcsMC4xMjE1OTI4NzMNClN0ZXJlb25lcGh0aHlhIHNwLixTQzE5MDAyMSwxMCw1LjUsLTIxLjQ2LDUuNTIsLTE4LjU3OCw4LjEwNTM5NDQwMSw0LjMyODUwNTM2OCwxNi4zODIyNTI1Niw4My42MTc3NDc0NCwzNjkyNTAwMDAsMTEzNy4wODMzMzMsLTI5LjM3NSw1MzAuODMzMzMzMywwLjQ2NjgzNzY2OSw0MjU3Ny4zNDg2OSwxMDMxNjQuMDk3LDE0NTc0MS40NDU3LDAuMDA2MTIyMzUzLDAuMTQzNjI3NDUxDQpMaXRvcGh5dG9uIHNwLixTQzE5MDAwNCwxMCw2LjI0LC0yMS44MTEsTkEsLTIwLjUwMSw4LjI3OTY2OTE4NyxOQSwyOC44NzAyOTI4OSw3MS4xMjk3MDcxMSwzMTA4NDA1NzkuNywyMjQ2LjIwMjg5OSwtNjYuMjg5ODU1MDcsNjczLjczOTEzMDQsMC4yOTk5NDU4MDIsMjU2NDMuNTEwODcsNDc5MjEuNjIyNjgsNzM1NjUuMTMzNTUsMC4wMTY5ODQ1MDUsMC4yMjk2MjY4DQpMaXRvcGh5dG9uIHNwLixTQzE5MDAwNSwxMCw2LjQxLC0yMS4zNTUsNi4wNzYsLTE5LjEzMyw4LjQyNTU4MTMxMiw2LjI3OTM2MTYwOSwyMS41MjQ2NjM2OCw3OC40NzUzMzYzMiwzMjYwODMzMzMuMywxOTcxLjI3MDgzMywtNjQuMjUsNTczLjA2MjUsMC4yOTA3MDcxMzcsMTA5NTg0LjkwMDEsNjczODcuMjM3MjksMTc2OTcyLjEzNzQsMC4wMTM0NzEzOTQsMC4yNDMzMzUzNDgNCkxpdG9waHl0b24gc3AuLFNDMTkwMDA2LDEwLDYuMjkzLC0yMC44MDYsNi4zMTQsLTE5LjU1Miw4LjEyODE2OTQ3NSw1Ljg4NTA0NjIwNCwyMi44MTU1MzM5OCw3Ny4xODQ0NjYwMiw0MDUxMDYzODMsMjI1NC44Mjk3ODcsLTY5LjU1MzE5MTQ5LDcwOS44OTM2MTcsMC4zMTQ4MzI0NjQsMTM4MzkxLjg1MDMsMTI0MjU5LjAyNTksMjYyNjUwLjg3NjIsMC4wMTI3MzE5MDgsMC4yNzQzNTQzOTMNCkxpdG9waHl0b24gc3AuLFNDMTkwMDA3LDEwLE5BLC0yMS42OTcsTkEsTkEsTkEsTkEsMjYuOTUzMTI1LDczLjA0Njg3NSwzLjA4RSswOCwxMTIwLjg2OTU2NSwtMi42NjY2NjY2NjcsMzI1Ljg1NTA3MjUsMC4yOTA3MTYzMTgsNjc1ODIuNzQ1MTksODk5MDUuMzE5NTcsMTU3NDg4LjA2NDgsMC4wMTUyOTk4NjgsMC4yNDMyMDc5MTYNCkxpdG9waHl0b24gc3AuLFNDMTkwMDA4LDEwLDYuMTI2LC0xOS41ODMsNi4wMDIsLTE3LjkyNyw4LjU3NzU2OTQxNiw2LjAzNTg2MDgyNywyOS4xNjY2NjY2Nyw3MC44MzMzMzMzMyw1MTYwNjM0OTIuMSwyMzE0LjY2NjY2NywtNjYuMTkwNDc2MTksNzQ0LjMxNzQ2MDMsMC4zMjE1NjU3MjMsNjM3MDcuNjM3ODIsODQ3NjUuOTk3MywxNDg0NzMuNjM1MSwwLjAxODE4MDU1NiwwLjI2MTkwNDc2Mg0KTGl0b3BoeXRvbiBzcC4sU0MxOTAwMDksMTAsNi4wMzMsLTIwLjU5OSxOQSwtMjAuOTksOC45MTE4ODE5MDIsTkEsMjcuNjE1MDYyNzYsNzIuMzg0OTM3MjQsMi44MEUrMDgsMTU1MS44NDg0ODUsLTI0Ljg0ODQ4NDg1LDUxMy4xMjEyMTIxLDAuMzMwNjUxNjE4LDM0MzI5Ljc3MjYyLDYzMjM3LjMxNTUsOTc1NjcuMDg4MTIsMC4wMTU2OTU0MjgsMC4yNDkxNzM0OTENCkxpdG9waHl0b24gc3AuLFNDMTkwMDE1LDEwLDUuNzgyLC0yMC4wMDIsTkEsLTE5Ljk5Nyw3LjU4NDk2NjcxOSxOQSwyNS4xOTA4Mzk2OSw3NC44MDkxNjAzMSwzMDcxNTE1MTUuMiwyMTA0LjMwMzAzLC0zNy42MjEyMTIxMiw2NjIuODc4Nzg3OSwwLjMxNTAxMTA4OCw0MDE2OS4zOTI5Niw5MTUyMy4xODk3NCwxMzE2OTIuNTgyNywwLjAxMjY3NTExMSwwLjI0MDE3MzUwMQ0KTGl0b3BoeXRvbiBzcC4sU0MxOTAwMTYsMTAsNS43MzQsLTIwLjc1NCw2LjAwMSwtMTcuNjUzLDguNDU5Mzk0OCw0LjE0NjM4MDA2NCwyNy43Nzc3Nzc3OCw3Mi4yMjIyMjIyMiwzNDQ4NjE1MzguNSwxMDg1LjIsLTE3LjQxNTM4NDYyLDM4NC43MDc2OTIzLDAuMzU0NTAzOTU1LDQzNTkwLjY4MTMsODI0OTMuMzgyMjMsMTI2MDg0LjA2MzUsMC4wMTMzNzEwMjEsMC4yNTY4NjA1OTMNCkxpdG9waHl0b24gc3AuLFNDMTkwMDE3LDEwLDUuNjU5LC0yMS43MzIsTkEsLTIxLjczMSw3Ljg1OTU2NzQwNixOQSwyOC4xMzg1MjgxNCw3MS44NjE0NzE4NiwzNjkxNjkyMzAuOCwxNjMwLjQxNTM4NSwtNTAuMjkyMzA3NjksNTEzLjMwNzY5MjMsMC4zMTQ4MzI0NjQsNzg3MjYuMDQxNTMsNDAxMTIuNDY2OTksMTE4ODM4LjUwODUsMC4wMTM2OTY4MzUsMC4yMTUyODcyNDQNCkxpdG9waHl0b24gc3AuLFNDMTkwMDIyLDEwLDUuODEzLC0xOC40OTksNS41NDgsLTE3LjE2OCw3LjY2MTkxOTE4Niw1LjI2OTU0MDczLDI2Ljk1MzEyNSw3My4wNDY4NzUsMjY1NzM5MTMwLjQsMTQyMC40OTI3NTQsLTQxLjY4MTE1OTQyLDQxNS45MTMwNDM1LDAuMjkyNzk0OTA3LDM1NjAxLjY0NjkzLDg0Nzk0LjMwNjE5LDEyMDM5NS45NTMxLDAuMDE2MjkyODc4LDAuMjg1ODEwNTgzDQo=" download="sf_res.csv">Download sf_res.csv</a>
<hr />
<p><strong>Statistics, the science of collecting, analyzing, presenting,
and interpreting data.</strong></p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span> (psych)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="fu">library</span>(gridExtra)</span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="fu">library</span>(car)</span></code></pre></div>
<p>Sample survey methods are used to collect data from observational
studies, and experimental design methods are used to collect data from
experimental studies. The area of descriptive statistics is concerned
primarily with methods of presenting and interpreting data using graphs,
tables, and numerical summaries. Whenever statisticians use data from a
sample—i.e., a subset of the population—to make statements about a
population, they are performing statistical inference. Estimation and
hypothesis testing are procedures used to make statistical inferences.
Probability plays a key role in statistical inference; it is used to
provide measures of the quality and precision of the inferences.</p>
<p>The data set <code>sf_res.csv</code> is extracted from Hsu et
al. (2020) and raw data are available [here] (<a
href="https://github.com/NTU-FRELab/stranger-things.git"
class="uri">https://github.com/NTU-FRELab/stranger-things.git</a>). The
data set describe organismal traits in two soft coral species (see
original paper for a full description of variable names, methods and
analyses.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>softies<span class="ot">&lt;-</span><span class="fu">read.csv</span>(<span class="st">&#39;data/sf_res.csv&#39;</span>, <span class="at">sep=</span><span class="st">&#39;,&#39;</span>, <span class="at">header=</span>T)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="fu">str</span>(softies)</span></code></pre></div>
<pre><code>## &#39;data.frame&#39;:    20 obs. of  21 variables:
##  $ species   : chr  &quot;Stereonephthya sp.&quot; &quot;Stereonephthya sp.&quot; &quot;Stereonephthya sp.&quot; &quot;Stereonephthya sp.&quot; ...
##  $ sample_no : chr  &quot;SC190001&quot; &quot;SC190003&quot; &quot;SC190010&quot; &quot;SC190012&quot; ...
##  $ depth     : int  10 10 10 10 10 10 10 10 10 10 ...
##  $ H15N      : num  NA 5.2 4.94 5.43 5.19 ...
##  $ H13C      : num  -23.2 -22.5 -22.2 -21.6 -21.2 ...
##  $ A15N      : num  5.58 5.43 4.98 5.53 5.38 ...
##  $ A13C      : num  -19.1 -19.3 -19.2 -18.9 -19.1 ...
##  $ H.C.N     : num  NA 6.73 6.52 6.36 8.19 ...
##  $ A.C.N     : num  4.18 5.08 4.63 5.53 5.01 ...
##  $ perc_om   : num  17 19 19.3 21 20 ...
##  $ perc_im   : num  83 81 80.7 79 80 ...
##  $ cell.afdw : num  5.54e+08 6.35e+08 3.94e+08 5.29e+08 4.09e+08 ...
##  $ chla.afdw : num  1112 1823 1063 1153 841 ...
##  $ chlb.afdw : num  -2.32 -24.415 -0.556 -1.525 -8.778 ...
##  $ chlc.afdw : num  540 715 483 558 389 ...
##  $ ratio     : num  0.486 0.392 0.454 0.484 0.462 ...
##  $ protZ.afdw: num  86426 82574 60822 53415 74083 ...
##  $ protH.afdw: num  55744 55068 38351 49313 46121 ...
##  $ prot.afdw : num  142170 137642 99173 102729 120204 ...
##  $ lip.cont  : num  0.00725 0.01057 0.00636 0.0117 0.00704 ...
##  $ lip.afdw  : num  0.166 0.152 0.123 0.208 0.164 ...</code></pre>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>softies<span class="sc">$</span>sample_no<span class="ot">&lt;-</span><span class="cn">NULL</span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>softies<span class="sc">$</span>species<span class="ot">&lt;-</span><span class="fu">as.factor</span>(softies<span class="sc">$</span>species)</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>softies<span class="sc">$</span>depth<span class="ot">&lt;-</span><span class="fu">as.factor</span>(softies<span class="sc">$</span>depth)</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="fu">str</span>(softies)</span></code></pre></div>
<pre><code>## &#39;data.frame&#39;:    20 obs. of  20 variables:
##  $ species   : Factor w/ 2 levels &quot;Litophyton sp.&quot;,..: 2 2 2 2 2 2 2 2 2 2 ...
##  $ depth     : Factor w/ 1 level &quot;10&quot;: 1 1 1 1 1 1 1 1 1 1 ...
##  $ H15N      : num  NA 5.2 4.94 5.43 5.19 ...
##  $ H13C      : num  -23.2 -22.5 -22.2 -21.6 -21.2 ...
##  $ A15N      : num  5.58 5.43 4.98 5.53 5.38 ...
##  $ A13C      : num  -19.1 -19.3 -19.2 -18.9 -19.1 ...
##  $ H.C.N     : num  NA 6.73 6.52 6.36 8.19 ...
##  $ A.C.N     : num  4.18 5.08 4.63 5.53 5.01 ...
##  $ perc_om   : num  17 19 19.3 21 20 ...
##  $ perc_im   : num  83 81 80.7 79 80 ...
##  $ cell.afdw : num  5.54e+08 6.35e+08 3.94e+08 5.29e+08 4.09e+08 ...
##  $ chla.afdw : num  1112 1823 1063 1153 841 ...
##  $ chlb.afdw : num  -2.32 -24.415 -0.556 -1.525 -8.778 ...
##  $ chlc.afdw : num  540 715 483 558 389 ...
##  $ ratio     : num  0.486 0.392 0.454 0.484 0.462 ...
##  $ protZ.afdw: num  86426 82574 60822 53415 74083 ...
##  $ protH.afdw: num  55744 55068 38351 49313 46121 ...
##  $ prot.afdw : num  142170 137642 99173 102729 120204 ...
##  $ lip.cont  : num  0.00725 0.01057 0.00636 0.0117 0.00704 ...
##  $ lip.afdw  : num  0.166 0.152 0.123 0.208 0.164 ...</code></pre>
<div id="descriptive" class="section level1">
<h1>Descriptive</h1>
<div id="all-in-one" class="section level2">
<h2>all-in-one</h2>
<p>Descriptive statistics are extracting basic information from a data
set. They are usually used to get a quick idea of possible differences,
data distribution, or just summarize them. They are easy to extract on a
single variable, but it may be a bit more complicated to get the same
information on multiple variables at once. If you are familiar with the
pivot table functionality in excel, the same can easily accomplished in
R such as here using the functions as <code>describe</code> and/or
<code>describeBy</code>. Let’s explore our <code>softies</code> data
set</p>
<ul>
<li>A quick <code>summary</code> of our data set in R (review):</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="fu">summary</span>(softies)</span></code></pre></div>
<pre><code>##                species   depth        H15N            H13C       
##  Litophyton sp.    :10   10:20   Min.   :4.943   Min.   :-23.22  
##  Stereonephthya sp.:10           1st Qu.:5.196   1st Qu.:-21.76  
##                                  Median :5.580   Median :-21.41  
##                                  Mean   :5.599   Mean   :-21.22  
##                                  3rd Qu.:5.978   3rd Qu.:-20.75  
##                                  Max.   :6.410   Max.   :-18.50  
##                                  NA&#39;s   :2                       
##       A15N            A13C            H.C.N           A.C.N      
##  Min.   :4.984   Min.   :-21.73   Min.   :5.735   Min.   :4.146  
##  1st Qu.:5.391   1st Qu.:-19.43   1st Qu.:6.944   1st Qu.:4.697  
##  Median :5.537   Median :-19.06   Median :8.117   Median :5.043  
##  Mean   :5.612   Mean   :-19.13   Mean   :7.755   Mean   :5.112  
##  3rd Qu.:5.911   3rd Qu.:-18.61   3rd Qu.:8.451   3rd Qu.:5.478  
##  Max.   :6.314   Max.   :-17.17   Max.   :9.419   Max.   :6.279  
##  NA&#39;s   :6       NA&#39;s   :1        NA&#39;s   :2       NA&#39;s   :6      
##     perc_om         perc_im        cell.afdw           chla.afdw     
##  Min.   :16.38   Min.   :70.83   Min.   :254222222   Min.   : 841.3  
##  1st Qu.:19.28   1st Qu.:72.88   1st Qu.:310130435   1st Qu.:1092.2  
##  Median :22.06   Median :77.94   Median :357015385   Median :1286.9  
##  Mean   :22.79   Mean   :77.21   Mean   :384159286   Mean   :1467.5  
##  3rd Qu.:27.12   3rd Qu.:80.72   3rd Qu.:415519126   3rd Qu.:1860.4  
##  Max.   :29.17   Max.   :83.62   Max.   :634716981   Max.   :2314.7  
##                                                                      
##    chlb.afdw        chlc.afdw         ratio          protZ.afdw    
##  Min.   :-69.55   Min.   :325.9   Min.   :0.2907   Min.   : 25644  
##  1st Qu.:-43.83   1st Qu.:449.9   1st Qu.:0.3148   1st Qu.: 43337  
##  Median :-24.63   Median :522.1   Median :0.3693   Median : 60943  
##  Mean   :-27.72   Mean   :535.5   Mean   :0.3815   Mean   : 65323  
##  3rd Qu.: -2.58   3rd Qu.:634.8   3rd Qu.:0.4631   3rd Qu.: 79688  
##  Max.   :  1.50   Max.   :744.3   Max.   :0.4857   Max.   :138392  
##                                                                    
##    protH.afdw       prot.afdw         lip.cont           lip.afdw     
##  Min.   : 34709   Min.   : 73565   Min.   :0.006122   Min.   :0.1048  
##  1st Qu.: 44694   1st Qu.:101840   1st Qu.:0.007200   1st Qu.:0.1503  
##  Median : 55977   Median :123240   Median :0.012186   Median :0.2115  
##  Mean   : 64659   Mean   :129982   Mean   :0.011320   Mean   :0.1998  
##  3rd Qu.: 84773   3rd Qu.:143063   3rd Qu.:0.014098   3rd Qu.:0.2448  
##  Max.   :124259   Max.   :262651   Max.   :0.018181   Max.   :0.2858  
## </code></pre>
<ul>
<li>A more detailed summary can be obtained using the function
<code>describe</code> from the package <code>psych</code>.</li>
</ul>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>psych<span class="sc">::</span><span class="fu">describe</span>(softies[,<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>)])</span></code></pre></div>
<pre><code>##            vars  n         mean           sd       median      trimmed
## H15N          1 18         5.60         0.48         5.58         5.59
## H13C          2 20       -21.22         1.05       -21.41       -21.28
## A15N          3 14         5.61         0.37         5.54         5.61
## A13C          4 19       -19.13         1.11       -19.06       -19.09
## H.C.N         5 18         7.76         1.06         8.12         7.78
## A.C.N         6 14         5.11         0.67         5.04         5.09
## perc_om       7 20        22.79         4.41        22.06        22.80
## perc_im       8 20        77.21         4.41        77.94        77.20
## cell.afdw     9 20 384159285.57 103344073.34 357015384.65 373386711.11
## chla.afdw    10 20      1467.55       488.05      1286.90      1436.14
## chlb.afdw    11 20       -27.72        24.88       -24.63       -26.35
## chlc.afdw    12 20       535.50       121.65       522.07       533.74
## ratio        13 20         0.38         0.08         0.37         0.38
## protZ.afdw   14 20     65323.05     27465.21     60942.54     62406.94
## protH.afdw   15 20     64659.06     25277.58     55976.61     62084.96
## prot.afdw    16 20    129982.12     39955.65    123240.01    124654.27
## lip.cont     17 20         0.01         0.00         0.01         0.01
## lip.afdw     18 20         0.20         0.06         0.21         0.20
##                    mad          min          max        range  skew kurtosis
## H15N              0.62         4.94         6.41         1.47  0.21    -1.46
## H13C              0.75       -23.22       -18.50         4.72  0.64     0.47
## A15N              0.24         4.98         6.31         1.33  0.30    -0.91
## A13C              0.72       -21.73       -17.17         4.56 -0.54    -0.05
## H.C.N             0.74         5.74         9.42         3.68 -0.52    -1.02
## A.C.N             0.66         4.15         6.28         2.13  0.15    -1.18
## perc_om           5.95        16.38        29.17        12.78  0.07    -1.59
## perc_im           5.95        70.83        83.62        12.78 -0.07    -1.59
## cell.afdw  73299191.06 254222222.20 634716981.10 380494758.90  0.88    -0.26
## chla.afdw       401.12       841.33      2314.67      1473.33  0.52    -1.33
## chlb.afdw        32.82       -69.55         1.50        71.05 -0.40    -1.31
## chlc.afdw       151.93       325.86       744.32       418.46  0.16    -1.19
## ratio             0.11         0.29         0.49         0.19  0.16    -1.80
## protZ.afdw    26797.03     25643.51    138391.85    112748.34  0.85     0.36
## protH.afdw    24826.08     34708.50    124259.03     89550.52  0.67    -0.68
## prot.afdw     31885.40     73565.13    262650.88    189085.74  1.66     3.50
## lip.cont          0.01         0.01         0.02         0.01  0.06    -1.54
## lip.afdw          0.07         0.10         0.29         0.18 -0.13    -1.55
##                     se
## H15N              0.11
## H13C              0.23
## A15N              0.10
## A13C              0.25
## H.C.N             0.25
## A.C.N             0.18
## perc_om           0.99
## perc_im           0.99
## cell.afdw  23108437.31
## chla.afdw       109.13
## chlb.afdw         5.56
## chlc.afdw        27.20
## ratio             0.02
## protZ.afdw     6141.41
## protH.afdw     5652.24
## prot.afdw      8934.36
## lip.cont          0.00
## lip.afdw          0.01</code></pre>
<ul>
<li>This is not very informative as we know we have two species.</li>
</ul>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a>psych<span class="sc">::</span><span class="fu">describe</span> (softies[,<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>)]<span class="sc">~</span>species)</span></code></pre></div>
<pre><code>## 
##  Descriptive statistics by group 
## group: Litophyton sp.
##            vars  n         mean          sd       median      trimmed
## species*      1 10         1.00        0.00         1.00         1.00
## depth*        2 10         1.00        0.00         1.00         1.00
## H15N          3  9         6.01        0.27         6.03         6.01
## H13C          4 10       -20.68        1.07       -20.78       -20.82
## A15N          5  5         5.99        0.28         6.00         5.99
## A13C          6  9       -19.41        1.57       -19.55       -19.41
## H.C.N         7  9         8.21        0.44         8.28         8.21
## A.C.N         8  5         5.52        0.86         5.89         5.52
## perc_om       9 10        26.50        2.55        27.28        26.79
## perc_im      10 10        73.50        2.55        72.72        73.21
## cell.afdw    11 10 343301520.30 73230471.56 318461956.50 331401572.56
## chla.afdw    12 10      1770.01      470.26      1800.84      1787.53
## chlb.afdw    13 10       -44.08       23.38       -45.99       -46.07
## chlc.afdw    14 10       551.68      145.35       543.19       555.83
## ratio        15 10         0.31        0.02         0.31         0.31
## protZ.afdw   16 10     63732.82    36489.94     53649.16     59161.60
## protH.afdw   17 10     77639.99    24143.29     83629.69     76503.55
## prot.afdw    18 10    141372.80    51723.52    128888.32    134689.00
## lip.cont     19 10         0.01        0.00         0.01         0.01
## lip.afdw     20 10         0.25        0.02         0.25         0.25
##                    mad          min          max        range  skew kurtosis
## species*          0.00         1.00         1.00         0.00   NaN      NaN
## depth*            0.00         1.00         1.00         0.00   NaN      NaN
## H15N              0.37         5.66         6.41         0.75  0.09    -1.80
## H13C              1.26       -21.81       -18.50         3.31  0.66    -0.83
## A15N              0.11         5.55         6.31         0.77 -0.47    -1.35
## A13C              2.13       -21.73       -17.17         4.56  0.05    -1.62
## H.C.N             0.44         7.58         8.91         1.33 -0.04    -1.47
## A.C.N             0.58         4.15         6.28         2.13 -0.64    -1.50
## perc_om           1.81        21.52        29.17         7.64 -0.81    -0.90
## perc_im           1.81        70.83        78.48         7.64  0.81    -0.90
## cell.afdw  48081858.49 265739130.40 516063492.10 250324361.70  1.16     0.35
## chla.afdw       612.10      1085.20      2314.67      1229.47 -0.21    -1.72
## chlb.afdw        30.03       -69.55        -2.67        66.89  0.40    -1.44
## chlc.afdw       191.13       325.86       744.32       418.46 -0.15    -1.63
## ratio             0.02         0.29         0.35         0.06  0.60    -0.71
## protZ.afdw    27700.08     25643.51    138391.85    112748.34  0.81    -0.77
## protH.afdw    17891.98     40112.47    124259.03     84146.56  0.17    -0.76
## prot.afdw     35719.58     73565.13    262650.88    189085.74  1.02     0.40
## lip.cont          0.00         0.01         0.02         0.01  0.34    -1.52
## lip.afdw          0.02         0.22         0.29         0.07  0.12    -1.02
##                     se
## species*          0.00
## depth*            0.00
## H15N              0.09
## H13C              0.34
## A15N              0.12
## A13C              0.52
## H.C.N             0.15
## A.C.N             0.38
## perc_om           0.81
## perc_im           0.81
## cell.afdw  23157508.43
## chla.afdw       148.71
## chlb.afdw         7.39
## chlc.afdw        45.96
## ratio             0.01
## protZ.afdw    11539.13
## protH.afdw     7634.78
## prot.afdw     16356.41
## lip.cont          0.00
## lip.afdw          0.01
## ------------------------------------------------------------ 
## group: Stereonephthya sp.
##            vars  n         mean           sd       median      trimmed
## species*      1 10         2.00         0.00         2.00         2.00
## depth*        2 10         1.00         0.00         1.00         1.00
## H15N          3  9         5.19         0.19         5.19         5.19
## H13C          4 10       -21.75         0.74       -21.67       -21.69
## A15N          5  9         5.40         0.21         5.43         5.40
## A13C          6 10       -18.88         0.32       -18.97       -18.90
## H.C.N         7  9         7.30         1.32         6.73         7.30
## A.C.N         8  9         4.88         0.44         4.98         4.88
## perc_om       9 10        19.08         1.99        19.28        18.97
## perc_im      10 10        80.92         1.99        80.72        81.03
## cell.afdw    11 10 425017050.84 116078754.42 401269841.30 420153913.14
## chla.afdw    12 10      1165.08       280.02      1103.28      1123.26
## chlb.afdw    13 10       -11.36        12.87        -5.55       -10.72
## chlc.afdw    14 10       519.32        97.63       506.77       511.17
## ratio        15 10         0.45         0.04         0.46         0.45
## protZ.afdw   16 10     66913.29     15979.83     60942.54     66841.24
## protH.afdw   17 10     51678.14     19787.75     47717.34     47363.60
## prot.afdw    18 10    118591.43     20165.34    118738.22    118523.38
## lip.cont     19 10         0.01         0.00         0.01         0.01
## lip.afdw     20 10         0.15         0.03         0.15         0.15
##                    mad          min          max        range  skew kurtosis
## species*          0.00         2.00         2.00         0.00   NaN      NaN
## depth*            0.00         1.00         1.00         0.00   NaN      NaN
## H15N              0.22         4.94         5.50         0.56  0.36    -1.45
## H13C              0.78       -23.22       -20.73         2.49 -0.50    -0.84
## A15N              0.14         4.98         5.64         0.66 -0.76    -0.72
## A13C              0.33       -19.30       -18.26         1.04  0.46    -1.11
## H.C.N             1.47         5.74         9.42         3.68  0.27    -1.70
## A.C.N             0.51         4.18         5.53         1.35 -0.22    -1.33
## perc_om           1.84        16.38        22.59         6.21  0.08    -1.20
## perc_im           1.84        77.41        83.62         6.21 -0.08    -1.20
## cell.afdw  97566037.45 254222222.20 634716981.10 380494758.90  0.37    -1.17
## chla.afdw       105.69       841.33      1823.45       982.12  1.19     0.39
## chlb.afdw        10.43       -29.38         1.50        30.88 -0.27    -1.95
## chlc.afdw        73.05       388.52       715.34       326.82  0.56    -0.78
## ratio             0.02         0.38         0.49         0.10 -0.79    -1.06
## protZ.afdw    15320.87     42577.35     91825.65     49248.30  0.21    -1.48
## protH.afdw    12245.20     34708.50    103164.10     68455.60  1.61     1.69
## prot.afdw     28517.35     91985.87    145741.45     53755.57  0.02    -1.82
## lip.cont          0.00         0.01         0.01         0.01  0.90    -0.76
## lip.afdw          0.03         0.10         0.21         0.10  0.31    -0.74
##                     se
## species*          0.00
## depth*            0.00
## H15N              0.06
## H13C              0.23
## A15N              0.07
## A13C              0.10
## H.C.N             0.44
## A.C.N             0.15
## perc_om           0.63
## perc_im           0.63
## cell.afdw  36707325.19
## chla.afdw        88.55
## chlb.afdw         4.07
## chlc.afdw        30.87
## ratio             0.01
## protZ.afdw     5053.27
## protH.afdw     6257.44
## prot.afdw      6376.84
## lip.cont          0.00
## lip.afdw          0.01</code></pre>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a>psych<span class="sc">::</span><span class="fu">describeBy</span> (softies[,<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>)],softies<span class="sc">$</span>species)</span></code></pre></div>
<pre><code>## 
##  Descriptive statistics by group 
## group: Litophyton sp.
##            vars  n         mean          sd       median      trimmed
## H15N          1  9         6.01        0.27         6.03         6.01
## H13C          2 10       -20.68        1.07       -20.78       -20.82
## A15N          3  5         5.99        0.28         6.00         5.99
## A13C          4  9       -19.41        1.57       -19.55       -19.41
## H.C.N         5  9         8.21        0.44         8.28         8.21
## A.C.N         6  5         5.52        0.86         5.89         5.52
## perc_om       7 10        26.50        2.55        27.28        26.79
## perc_im       8 10        73.50        2.55        72.72        73.21
## cell.afdw     9 10 343301520.30 73230471.56 318461956.50 331401572.56
## chla.afdw    10 10      1770.01      470.26      1800.84      1787.53
## chlb.afdw    11 10       -44.08       23.38       -45.99       -46.07
## chlc.afdw    12 10       551.68      145.35       543.19       555.83
## ratio        13 10         0.31        0.02         0.31         0.31
## protZ.afdw   14 10     63732.82    36489.94     53649.16     59161.60
## protH.afdw   15 10     77639.99    24143.29     83629.69     76503.55
## prot.afdw    16 10    141372.80    51723.52    128888.32    134689.00
## lip.cont     17 10         0.01        0.00         0.01         0.01
## lip.afdw     18 10         0.25        0.02         0.25         0.25
##                    mad          min          max        range  skew kurtosis
## H15N              0.37         5.66         6.41         0.75  0.09    -1.80
## H13C              1.26       -21.81       -18.50         3.31  0.66    -0.83
## A15N              0.11         5.55         6.31         0.77 -0.47    -1.35
## A13C              2.13       -21.73       -17.17         4.56  0.05    -1.62
## H.C.N             0.44         7.58         8.91         1.33 -0.04    -1.47
## A.C.N             0.58         4.15         6.28         2.13 -0.64    -1.50
## perc_om           1.81        21.52        29.17         7.64 -0.81    -0.90
## perc_im           1.81        70.83        78.48         7.64  0.81    -0.90
## cell.afdw  48081858.49 265739130.40 516063492.10 250324361.70  1.16     0.35
## chla.afdw       612.10      1085.20      2314.67      1229.47 -0.21    -1.72
## chlb.afdw        30.03       -69.55        -2.67        66.89  0.40    -1.44
## chlc.afdw       191.13       325.86       744.32       418.46 -0.15    -1.63
## ratio             0.02         0.29         0.35         0.06  0.60    -0.71
## protZ.afdw    27700.08     25643.51    138391.85    112748.34  0.81    -0.77
## protH.afdw    17891.98     40112.47    124259.03     84146.56  0.17    -0.76
## prot.afdw     35719.58     73565.13    262650.88    189085.74  1.02     0.40
## lip.cont          0.00         0.01         0.02         0.01  0.34    -1.52
## lip.afdw          0.02         0.22         0.29         0.07  0.12    -1.02
##                     se
## H15N              0.09
## H13C              0.34
## A15N              0.12
## A13C              0.52
## H.C.N             0.15
## A.C.N             0.38
## perc_om           0.81
## perc_im           0.81
## cell.afdw  23157508.43
## chla.afdw       148.71
## chlb.afdw         7.39
## chlc.afdw        45.96
## ratio             0.01
## protZ.afdw    11539.13
## protH.afdw     7634.78
## prot.afdw     16356.41
## lip.cont          0.00
## lip.afdw          0.01
## ------------------------------------------------------------ 
## group: Stereonephthya sp.
##            vars  n         mean           sd       median      trimmed
## H15N          1  9         5.19         0.19         5.19         5.19
## H13C          2 10       -21.75         0.74       -21.67       -21.69
## A15N          3  9         5.40         0.21         5.43         5.40
## A13C          4 10       -18.88         0.32       -18.97       -18.90
## H.C.N         5  9         7.30         1.32         6.73         7.30
## A.C.N         6  9         4.88         0.44         4.98         4.88
## perc_om       7 10        19.08         1.99        19.28        18.97
## perc_im       8 10        80.92         1.99        80.72        81.03
## cell.afdw     9 10 425017050.84 116078754.42 401269841.30 420153913.14
## chla.afdw    10 10      1165.08       280.02      1103.28      1123.26
## chlb.afdw    11 10       -11.36        12.87        -5.55       -10.72
## chlc.afdw    12 10       519.32        97.63       506.77       511.17
## ratio        13 10         0.45         0.04         0.46         0.45
## protZ.afdw   14 10     66913.29     15979.83     60942.54     66841.24
## protH.afdw   15 10     51678.14     19787.75     47717.34     47363.60
## prot.afdw    16 10    118591.43     20165.34    118738.22    118523.38
## lip.cont     17 10         0.01         0.00         0.01         0.01
## lip.afdw     18 10         0.15         0.03         0.15         0.15
##                    mad          min          max        range  skew kurtosis
## H15N              0.22         4.94         5.50         0.56  0.36    -1.45
## H13C              0.78       -23.22       -20.73         2.49 -0.50    -0.84
## A15N              0.14         4.98         5.64         0.66 -0.76    -0.72
## A13C              0.33       -19.30       -18.26         1.04  0.46    -1.11
## H.C.N             1.47         5.74         9.42         3.68  0.27    -1.70
## A.C.N             0.51         4.18         5.53         1.35 -0.22    -1.33
## perc_om           1.84        16.38        22.59         6.21  0.08    -1.20
## perc_im           1.84        77.41        83.62         6.21 -0.08    -1.20
## cell.afdw  97566037.45 254222222.20 634716981.10 380494758.90  0.37    -1.17
## chla.afdw       105.69       841.33      1823.45       982.12  1.19     0.39
## chlb.afdw        10.43       -29.38         1.50        30.88 -0.27    -1.95
## chlc.afdw        73.05       388.52       715.34       326.82  0.56    -0.78
## ratio             0.02         0.38         0.49         0.10 -0.79    -1.06
## protZ.afdw    15320.87     42577.35     91825.65     49248.30  0.21    -1.48
## protH.afdw    12245.20     34708.50    103164.10     68455.60  1.61     1.69
## prot.afdw     28517.35     91985.87    145741.45     53755.57  0.02    -1.82
## lip.cont          0.00         0.01         0.01         0.01  0.90    -0.76
## lip.afdw          0.03         0.10         0.21         0.10  0.31    -0.74
##                     se
## H15N              0.06
## H13C              0.23
## A15N              0.07
## A13C              0.10
## H.C.N             0.44
## A.C.N             0.15
## perc_om           0.63
## perc_im           0.63
## cell.afdw  36707325.19
## chla.afdw        88.55
## chlb.afdw         4.07
## chlc.afdw        30.87
## ratio             0.01
## protZ.afdw     5053.27
## protH.afdw     6257.44
## prot.afdw      6376.84
## lip.cont          0.00
## lip.afdw          0.01</code></pre>
</div>
<div id="counts-proportions" class="section level2">
<h2>Counts &amp; proportions</h2>
<p>Simple tabulation (count) summary are made
using<code>table</code></p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="co"># One variables</span></span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a><span class="fu">prop.table</span> (<span class="fu">table</span>(softies<span class="sc">$</span>species))</span></code></pre></div>
<pre><code>## 
##     Litophyton sp. Stereonephthya sp. 
##                0.5                0.5</code></pre>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a><span class="co"># two variables</span></span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a><span class="fu">table</span>(softies<span class="sc">$</span>depth, softies<span class="sc">$</span>species)</span></code></pre></div>
<pre><code>##     
##      Litophyton sp. Stereonephthya sp.
##   10             10                 10</code></pre>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a><span class="fu">prop.table</span> (<span class="fu">table</span>(softies<span class="sc">$</span>depth, softies<span class="sc">$</span>species))</span></code></pre></div>
<pre><code>##     
##      Litophyton sp. Stereonephthya sp.
##   10            0.5                0.5</code></pre>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a><span class="co"># three variables, with nicer formatting after adding a random variables</span></span>
<span id="cb20-2"><a href="#cb20-2" tabindex="-1"></a>softies_2<span class="ot">&lt;-</span>softies</span>
<span id="cb20-3"><a href="#cb20-3" tabindex="-1"></a>softies_2<span class="sc">$</span>site<span class="ot">&lt;-</span><span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;B&quot;</span>),<span class="dv">10</span>)</span>
<span id="cb20-4"><a href="#cb20-4" tabindex="-1"></a><span class="fu">ftable</span>(softies_2<span class="sc">$</span>depth, softies_2<span class="sc">$</span>species, softies_2<span class="sc">$</span>site)</span></code></pre></div>
<pre><code>##                        A B
##                           
## 10 Litophyton sp.      5 5
##    Stereonephthya sp.  5 5</code></pre>
</div>
<div id="aggregate-apply" class="section level2">
<h2><code>aggregate</code> &amp; <code>apply</code></h2>
<p>We previously applied a diversity of function on a selected vector,
created a filter and applied it on subgroups. In <code>softies</code>,
the ratio represents the amount of chl c / chl a in the samples. It is
informing on the use of light.</p>
<p><img src="stat_1_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a><span class="fu">mean</span>(softies<span class="sc">$</span>ratio)</span></code></pre></div>
<pre><code>## [1] 0.3814767</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a>lit <span class="ot">&lt;-</span> softies<span class="sc">$</span>species <span class="sc">==</span> <span class="st">&#39;Litophyton sp.&#39;</span></span>
<span id="cb24-2"><a href="#cb24-2" tabindex="-1"></a><span class="fu">mean</span>(softies<span class="sc">$</span>ratio[lit])</span></code></pre></div>
<pre><code>## [1] 0.3125561</code></pre>
<p>Descriptive statistics can be calculated on subgroups using
<code>aggregate</code> and <code>apply</code> functions. The difference
between the two is that the 2nd argument in <code>aggregate</code> must
be a list while in <code>tapply</code> it can be a list but this is not
mandatory. As a consequence, the output of <code>aggregate</code> is a
data frame while the one of <code>tapply</code> is an array (basically,
you deal with the list before or after).</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a><span class="fu">aggregate</span>(softies<span class="sc">$</span>ratio,<span class="fu">list</span> (<span class="at">species=</span>softies<span class="sc">$</span>species),<span class="at">FUN=</span><span class="st">&quot;median&quot;</span>)</span></code></pre></div>
<pre><code>##              species         x
## 1     Litophyton sp. 0.3148325
## 2 Stereonephthya sp. 0.4643134</code></pre>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" tabindex="-1"></a><span class="fu">tapply</span>(softies<span class="sc">$</span>ratio,softies<span class="sc">$</span>species, <span class="at">FUN=</span><span class="st">&quot;median&quot;</span>)</span></code></pre></div>
<pre><code>##     Litophyton sp. Stereonephthya sp. 
##          0.3148325          0.4643134</code></pre>
<p>The wide family of <code>apply</code> functions represents common
alternatives to <strong>loops</strong>. You can read more about the wide
application of these functions in many online resources.</p>
<p class="comment">
<strong>Practice 7.1</strong> Using the data <code>iris</code> explore
how the four traits vary among flower species. (1) Use
<code>boxplot</code> as visual summaries then extract descriptive
statistics by species (<code>describeBy</code> or others) (2) Count
number of observations in each trait and species (it should result in a
table of three species by four traits filled with 50 values) (3)
Calculate the <code>median</code> of each variable by
<code>Species</code>, then calculate the <code>mean</code> by
<code>Species</code> but for the <code>Sepal.Length</code> only.
</p>
<div class="sourceCode" id="cb30"><pre
class="sourceCode r fold-hide"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" tabindex="-1"></a>plot1 <span class="ot">&lt;-</span><span class="fu">ggplot</span>(iris, <span class="fu">aes</span>(<span class="at">x=</span>Species, <span class="at">y=</span>Sepal.Length)) <span class="sc">+</span> </span>
<span id="cb30-2"><a href="#cb30-2" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()</span>
<span id="cb30-3"><a href="#cb30-3" tabindex="-1"></a>  </span>
<span id="cb30-4"><a href="#cb30-4" tabindex="-1"></a>plot2 <span class="ot">&lt;-</span><span class="fu">ggplot</span>(iris, <span class="fu">aes</span>(<span class="at">x=</span>Species, <span class="at">y=</span>Sepal.Width)) <span class="sc">+</span> </span>
<span id="cb30-5"><a href="#cb30-5" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() </span>
<span id="cb30-6"><a href="#cb30-6" tabindex="-1"></a>  </span>
<span id="cb30-7"><a href="#cb30-7" tabindex="-1"></a>plot3 <span class="ot">&lt;-</span><span class="fu">ggplot</span>(iris, <span class="fu">aes</span>(<span class="at">x=</span>Species, <span class="at">y=</span>Petal.Length)) <span class="sc">+</span> </span>
<span id="cb30-8"><a href="#cb30-8" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() </span>
<span id="cb30-9"><a href="#cb30-9" tabindex="-1"></a>  </span>
<span id="cb30-10"><a href="#cb30-10" tabindex="-1"></a>plot4 <span class="ot">&lt;-</span><span class="fu">ggplot</span>(iris, <span class="fu">aes</span>(<span class="at">x=</span>Species, <span class="at">y=</span>Petal.Width)) <span class="sc">+</span> </span>
<span id="cb30-11"><a href="#cb30-11" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() </span>
<span id="cb30-12"><a href="#cb30-12" tabindex="-1"></a>  </span>
<span id="cb30-13"><a href="#cb30-13" tabindex="-1"></a><span class="fu">grid.arrange</span>(plot1, plot2,plot3, plot4, <span class="at">ncol=</span><span class="dv">2</span>)</span>
<span id="cb30-14"><a href="#cb30-14" tabindex="-1"></a><span class="fu">describeBy</span> (iris, iris<span class="sc">$</span>Species)</span>
<span id="cb30-15"><a href="#cb30-15" tabindex="-1"></a></span>
<span id="cb30-16"><a href="#cb30-16" tabindex="-1"></a>iris <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Species) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="fu">across</span>(<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>), length))</span>
<span id="cb30-17"><a href="#cb30-17" tabindex="-1"></a><span class="fu">aggregate</span>(iris[,<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>],<span class="at">by=</span><span class="fu">list</span>(iris<span class="sc">$</span>Species), median)</span>
<span id="cb30-18"><a href="#cb30-18" tabindex="-1"></a><span class="fu">tapply</span>(iris<span class="sc">$</span>Sepal.Length , iris<span class="sc">$</span>Species, mean)</span></code></pre></div>
</div>
</div>
<div id="hypothesis-testing" class="section level1">
<h1>Hypothesis testing</h1>
<p>One of the most basic concepts in statistics is hypothesis testing
and something called <strong>The Null Hypothesis</strong>. This class is
not about statitics but some basic concepts and understanding are
needed. The amazing YouTube Channel <a
href="https://www.youtube.com/c/joshstarmer">StatQuest</a> can help you
with that. Josh Starmer is a professor and <strong>a musical
artist</strong> at University of North Carolina at Chapel Hill. You have
hours of <a
href="https://youtube.com/playlist?list=PLblh5JKOoLUK0FLuzwntyYI10UQFUhsY9">enjoyment</a>
to listen music and learn about stats. Today, let’s got to the
fundamental of statistics and explanation about [The Null
Hypothesis]((<a
href="https://www.youtube.com/watch?v=0oc49DyA3hU&amp;list=PLblh5JKOoLUK0FLuzwntyYI10UQFUhsY9&amp;index=8"
class="uri">https://www.youtube.com/watch?v=0oc49DyA3hU&amp;list=PLblh5JKOoLUK0FLuzwntyYI10UQFUhsY9&amp;index=8</a>)
is a perfect introduction to the topic.</p>
<p>The correct formulation of the <strong>hypotheses</strong> for what
you want to test is crucial: <em>H0</em> stands for the null hypothesis
- no difference is detected (does not mean there is no difference),
<em>H1</em> stands for the alternative hypothesis - the presence of a
difference.</p>
<p>Two broad categories of statistics can be distinguished parametric
(assuming normal distribution &amp; homoscedasticity - video 3) and non
parametric (no assumption of normality).</p>
<p>Note: In these tests, it is important to understand the concept of
degrees of freedom: It refers to <strong>the maximum number of logically
independent values</strong>, i.e. values that have the freedom to vary
in the data sample. Look <a
href="https://www.youtube.com/watch?v=N20rl2llHno&amp;ab_channel=zedstatistics">here</a>
if you are not clear with this term, as it is the foundation you will
need later on.</p>
<div class="line-block">Simply, to calculate the degrees of freedom for
a sample mean, you can use the formula <span
class="math inline">\(df=n-1\)</span>, where <span
class="math inline">\(n\)</span> is the sample size as we just only one
number will determine the mean; the others have the freedom to
vary.</div>
<div id="correlations" class="section level2">
<h2>Correlations</h2>
<p>Let’s first examine what is a correlation.</p>
<ul>
<li><p><em>Pearson</em> correlation - parametric</p>
<ul>
<li><span class="math inline">\(r\)</span> pearson coefficient <span
class="math display">\[
r = \frac{{}\sum_{i=1}^{n} (x_i - \overline{x})(y_i - \overline{y})}
{\sqrt{\sum_{i=1}^{n} (x_i - \overline{x})^2(y_i - \overline{y})^2}}
\]</span></li>
<li>Formula for the <span class="math inline">\(t\)</span> test fro the
correlation coefficient: <span class="math display">\[
t = r\sqrt\frac{n-2}{1-r^2}
\]</span> ___________with degrees of freedom equal to <span
class="math inline">\(n-2\)</span></li>
</ul></li>
</ul>
<p>It may look scary for some of you but it is actually very easy to
compute (see a demo in step-by step demo in excel <a
href=".\Data\demo_stats.xlsx">here</a>).</p>
<ul>
<li>Example:</li>
</ul>
<div class="sourceCode" id="cb31"><pre
class="sourceCode r fold-show"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" tabindex="-1"></a><span class="co"># dataset hypotheses?</span></span>
<span id="cb31-2"><a href="#cb31-2" tabindex="-1"></a>x<span class="ot">&lt;-</span>students<span class="sc">$</span>height</span>
<span id="cb31-3"><a href="#cb31-3" tabindex="-1"></a>y<span class="ot">&lt;-</span>students<span class="sc">$</span>shoesize</span>
<span id="cb31-4"><a href="#cb31-4" tabindex="-1"></a>s<span class="ot">&lt;-</span>students[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>] <span class="co"># a matrix</span></span>
<span id="cb31-5"><a href="#cb31-5" tabindex="-1"></a><span class="co"># Pearson correlation</span></span>
<span id="cb31-6"><a href="#cb31-6" tabindex="-1"></a><span class="co"># cor(x,y)</span></span>
<span id="cb31-7"><a href="#cb31-7" tabindex="-1"></a><span class="co"># cor(s)</span></span>
<span id="cb31-8"><a href="#cb31-8" tabindex="-1"></a><span class="fu">cor.test</span>(x,y)</span></code></pre></div>
<p>This relationship is usually represented with it confident interval
on a scatter plot.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" tabindex="-1"></a><span class="fu">ggplot</span>(students, <span class="fu">aes</span>(<span class="at">x =</span> height, <span class="at">y =</span> shoesize)) <span class="sc">+</span> </span>
<span id="cb32-2"><a href="#cb32-2" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb32-3"><a href="#cb32-3" tabindex="-1"></a>  <span class="fu">stat_smooth</span>(<span class="at">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="at">col =</span> <span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<ul>
<li><em>Spearman</em> correlation - non-parametric</li>
</ul>
<p>It is based on ranking, and tests the existence of a monotonic
relationship.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" tabindex="-1"></a><span class="co"># Spearman correlation (monotonic)</span></span>
<span id="cb33-2"><a href="#cb33-2" tabindex="-1"></a><span class="co"># cor(x,y, method =&#39;spearman&#39;)</span></span>
<span id="cb33-3"><a href="#cb33-3" tabindex="-1"></a><span class="fu">cor.test</span>(x,y, <span class="at">method =</span><span class="st">&#39;spearman&#39;</span>)</span></code></pre></div>
<p><strong>The example below illustrates on the importance of running
both types of correlation</strong></p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" tabindex="-1"></a>w<span class="ot">&lt;-</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>)</span>
<span id="cb34-2"><a href="#cb34-2" tabindex="-1"></a>z<span class="ot">&lt;-</span><span class="fu">exp</span>(x)</span>
<span id="cb34-3"><a href="#cb34-3" tabindex="-1"></a><span class="fu">cor.test</span>(w,z,<span class="at">method=</span><span class="st">&#39;pearson&#39;</span>) <span class="co"># super low</span></span>
<span id="cb34-4"><a href="#cb34-4" tabindex="-1"></a><span class="fu">cor.test</span>(w,z,<span class="at">method=</span><span class="st">&#39;spearman&#39;</span>) <span class="co">#super high</span></span></code></pre></div>
<p class="comment">
<strong>Practice 7.2</strong> Play with modifying a value from the
student data set and check how <span class="math inline">\(t\)</span>
and <span class="math inline">\(p\)</span> values are affected by this
change. Examine change in the confidence interval.
</p>
</div>
<div id="chi-square-test" class="section level2">
<h2>Chi-square test</h2>
<p>It is a goodness-of-fit test: observed vs theoretical.</p>
<ul>
<li>Casino example</li>
</ul>
<center>
<img src="Figures/die.png" style="width:50.0%" />
</center>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" tabindex="-1"></a><span class="co">#Cast 240 times a die. We counted occurrence of 1,2,3,4,5,6</span></span>
<span id="cb35-2"><a href="#cb35-2" tabindex="-1"></a>die<span class="ot">&lt;-</span><span class="fu">data.frame</span>(<span class="at">obs=</span><span class="fu">c</span>(<span class="dv">55</span>,<span class="dv">44</span>,<span class="dv">35</span>,<span class="dv">45</span>,<span class="dv">31</span>,<span class="dv">30</span>), <span class="at">row.names=</span><span class="fu">c</span>(<span class="st">&#39;Cast1&#39;</span>,<span class="st">&#39;Cast2&#39;</span>,<span class="st">&#39;Cast3&#39;</span>,<span class="st">&#39;Cast4&#39;</span>,<span class="st">&#39;Cast5&#39;</span>,<span class="st">&#39;Cast6&#39;</span>))</span>
<span id="cb35-3"><a href="#cb35-3" tabindex="-1"></a>die <span class="co">#Is this die fair? Define H0 and H1.  </span></span></code></pre></div>
<p>Assumption: all results are equally probably (H0, uniform
distribution).</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" tabindex="-1"></a><span class="fu">chisq.test</span>(die)</span>
<span id="cb36-2"><a href="#cb36-2" tabindex="-1"></a><span class="co"># I am cheating</span></span></code></pre></div>
<ul>
<li>Hardy-Weinberg</li>
</ul>
<p>A common biological application of a Chi-square test is the
Hardy-Weinberg equilibrium. Genotypes at the Hardy-Weinberg equilibrium
will follow: <em>p</em>2 + 2<em>pq</em> + <em>q</em>2 = 1 at panmixia
(random mating, <em>p</em> and <em>q</em> represent allele frequency).
In this case, we want to compare our observation to a theoretical
distribution.</p>
<p>Let’s assume 2 alleles (<em>A</em> and <em>T</em>) with observed
genotypes in a population: 750 <em>AA</em>, <em>50</em> AT, 200
<em>TT</em> =&gt; <em>f(A)</em>: 0.775 / <em>f(T)</em>= 0.225.
Theoretical distribution should follow: 0.60(<em>p2</em>), 0.5
(<em>2pq</em>), 0.05 (<em>q2</em>) with <em>p2</em> + <em>2pq</em> +
<em>q2</em> = 1. Is our population at the HW-equilibrium (H0)?</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" tabindex="-1"></a>obs <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">750</span>, <span class="dv">50</span>, <span class="dv">200</span>)</span>
<span id="cb37-2"><a href="#cb37-2" tabindex="-1"></a>exp <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.60</span>, <span class="fl">0.35</span>, <span class="fl">0.05</span>)</span>
<span id="cb37-3"><a href="#cb37-3" tabindex="-1"></a><span class="fu">chisq.test</span> (<span class="at">x=</span>obs, <span class="at">p=</span>exp)</span></code></pre></div>
<ul>
<li>Frequencies</li>
</ul>
<p>Chi-square test can simply be used to compare the distribution of
frequencies in two populations:</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" tabindex="-1"></a>F <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="at">nrow=</span><span class="dv">4</span>,<span class="at">ncol=</span><span class="dv">2</span>,<span class="at">data=</span><span class="fu">c</span>(<span class="dv">33</span>,<span class="dv">14</span>, <span class="dv">8</span>,<span class="dv">18</span>,<span class="dv">31</span>,<span class="dv">25</span>,<span class="dv">14</span>,<span class="dv">12</span>))</span>
<span id="cb38-2"><a href="#cb38-2" tabindex="-1"></a><span class="fu">chisq.test</span>(F) <span class="co"># alternative see `fisher.test`</span></span></code></pre></div>
</div>
<div id="student-t-test" class="section level2">
<h2>Student t-test</h2>
<p>Several version of t-test exist: one sample compared to a known mean,
two samples (with variance equal - Student’s or with unequal variance -
Welsh’s), <strong>paired</strong> (compare two dependent
(e.g. before-after) samples. See also the step-by step demo in excel <a
href=".\Data\demo_stats.xlsx">here</a>)</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" tabindex="-1"></a><span class="co"># One sample</span></span>
<span id="cb39-2"><a href="#cb39-2" tabindex="-1"></a><span class="fu">t.test</span> (students<span class="sc">$</span>height, <span class="at">mu=</span><span class="dv">170</span>)</span>
<span id="cb39-3"><a href="#cb39-3" tabindex="-1"></a><span class="co"># Two sample (with equal variances)</span></span>
<span id="cb39-4"><a href="#cb39-4" tabindex="-1"></a><span class="fu">t.test</span> (students<span class="sc">$</span>height<span class="sc">~</span>students<span class="sc">$</span>gender, <span class="at">var.equal =</span> <span class="cn">TRUE</span>)</span>
<span id="cb39-5"><a href="#cb39-5" tabindex="-1"></a><span class="co"># Two sample (with unequal variances, default option when using t.test) </span></span>
<span id="cb39-6"><a href="#cb39-6" tabindex="-1"></a><span class="fu">t.test</span> (students<span class="sc">$</span>height<span class="sc">~</span>students<span class="sc">$</span>gender)</span>
<span id="cb39-7"><a href="#cb39-7" tabindex="-1"></a><span class="co"># Two sample paired t.test</span></span>
<span id="cb39-8"><a href="#cb39-8" tabindex="-1"></a><span class="fu">t.test</span> (students<span class="sc">$</span>height<span class="sc">~</span>students<span class="sc">$</span>gender, <span class="at">paired=</span>T)</span></code></pre></div>
<p>In the case of a two samples assuming equal variances, the <em>t</em>
statistic is obtained as following:</p>
<p><img src="Figures/ttest.png" /></p>
<p class="comment">
<strong>Practice 7.3</strong> Using <code>iris</code>. Does
<code>Sepal.Lenght</code> differ between species <code>setosa</code> and
<code>versicolor</code>? Does <code>Sepal.Lenght</code> differ between
<code>virginica</code> and <code>versicolor</code>? Make a plot, define
your hypotheses, test hypotheses
</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" tabindex="-1"></a>set <span class="ot">&lt;-</span> iris[iris<span class="sc">$</span>Species <span class="sc">==</span> <span class="st">&quot;setosa&quot;</span>, ]<span class="sc">$</span>Sepal.Length</span>
<span id="cb40-2"><a href="#cb40-2" tabindex="-1"></a>ver <span class="ot">&lt;-</span> iris[iris<span class="sc">$</span>Species <span class="sc">==</span> <span class="st">&quot;versicolor&quot;</span>, ]<span class="sc">$</span>Sepal.Length</span>
<span id="cb40-3"><a href="#cb40-3" tabindex="-1"></a>vir <span class="ot">&lt;-</span> iris[iris<span class="sc">$</span>Species <span class="sc">==</span> <span class="st">&quot;virginica&quot;</span>, ]<span class="sc">$</span>Sepal.Length</span>
<span id="cb40-4"><a href="#cb40-4" tabindex="-1"></a></span>
<span id="cb40-5"><a href="#cb40-5" tabindex="-1"></a>setver <span class="ot">&lt;-</span> <span class="fu">t.test</span>(set, ver, <span class="at">paired =</span> <span class="cn">FALSE</span>, <span class="at">alternative =</span> <span class="st">&quot;two.sided&quot;</span>, <span class="at">var.equal =</span> <span class="cn">FALSE</span>)</span>
<span id="cb40-6"><a href="#cb40-6" tabindex="-1"></a>vervir <span class="ot">&lt;-</span> <span class="fu">t.test</span>(ver, vir, <span class="at">paired =</span> <span class="cn">FALSE</span>, <span class="at">alternative =</span> <span class="st">&quot;two.sided&quot;</span>, <span class="at">var.equal =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</div>
<div id="mann-whitney" class="section level2">
<h2>Mann-Whitney</h2>
<p>A non-parametric solution for the comparison of two samples:
Mann-Whitney U-test (independent) or Wilcoxon W-test (dependant)</p>
<div class="sourceCode" id="cb41"><pre
class="sourceCode r fold-show"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" tabindex="-1"></a><span class="co"># Normality plot &amp; test</span></span>
<span id="cb41-2"><a href="#cb41-2" tabindex="-1"></a>students<span class="sc">$</span>height[<span class="dv">6</span>]<span class="ot">&lt;-</span><span class="dv">132</span></span>
<span id="cb41-3"><a href="#cb41-3" tabindex="-1"></a>students<span class="sc">$</span>height[<span class="dv">10</span>]<span class="ot">&lt;-</span><span class="dv">310</span></span>
<span id="cb41-4"><a href="#cb41-4" tabindex="-1"></a>students<span class="sc">$</span>height[<span class="dv">8</span>]<span class="ot">&lt;-</span><span class="dv">132</span></span>
<span id="cb41-5"><a href="#cb41-5" tabindex="-1"></a>students<span class="sc">$</span>height[<span class="dv">9</span>]<span class="ot">&lt;-</span><span class="dv">210</span></span>
<span id="cb41-6"><a href="#cb41-6" tabindex="-1"></a><span class="fu">boxplot</span>(height<span class="sc">~</span>gender, students)</span>
<span id="cb41-7"><a href="#cb41-7" tabindex="-1"></a><span class="fu">qqnorm</span>(students<span class="sc">$</span>height) </span>
<span id="cb41-8"><a href="#cb41-8" tabindex="-1"></a><span class="fu">qqline</span>(students<span class="sc">$</span>height) </span>
<span id="cb41-9"><a href="#cb41-9" tabindex="-1"></a><span class="fu">shapiro.test</span>(students<span class="sc">$</span>height) <span class="co"># data are not normal </span></span>
<span id="cb41-10"><a href="#cb41-10" tabindex="-1"></a><span class="fu">wilcox.test</span> (students<span class="sc">$</span>height<span class="sc">~</span>students<span class="sc">$</span>gender) </span></code></pre></div>
</div>
<div id="variances" class="section level2">
<h2>Variances</h2>
<p>Variance tests are used to determine if the variances of two (or
more) populations are equal. F-test are generally very sensitive to the
non-normality of the data as you can imagine from the previous
<code>boxplot</code>. The <strong>homoscedasticity</strong> (or
<strong>homogeneity of variance</strong>) is an important condition in
parametric statistics.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" tabindex="-1"></a><span class="co"># Test of variance: we test HO: homogeneous, H1:heterogeneous</span></span>
<span id="cb42-2"><a href="#cb42-2" tabindex="-1"></a><span class="fu">fligner.test</span> (students<span class="sc">$</span>height <span class="sc">~</span> students<span class="sc">$</span>gender)</span></code></pre></div>
<p>Another example using the <code>ToothGrowth</code> data set
(<code>car</code> package) on multiple groups</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" tabindex="-1"></a>tg<span class="ot">&lt;-</span>ToothGrowth</span>
<span id="cb43-2"><a href="#cb43-2" tabindex="-1"></a>tg<span class="sc">$</span>dose<span class="ot">&lt;-</span><span class="fu">factor</span>(tg<span class="sc">$</span>dose)</span>
<span id="cb43-3"><a href="#cb43-3" tabindex="-1"></a><span class="fu">boxplot</span>(len<span class="sc">~</span>dose<span class="sc">*</span>supp, <span class="at">data=</span>tg)</span>
<span id="cb43-4"><a href="#cb43-4" tabindex="-1"></a><span class="co"># also work with: boxplot(len ~ interaction (dose,supp), data=tg)</span></span>
<span id="cb43-5"><a href="#cb43-5" tabindex="-1"></a><span class="co"># or: plot(len ~ interaction (dose,supp), data=tg)</span></span>
<span id="cb43-6"><a href="#cb43-6" tabindex="-1"></a><span class="fu">bartlett.test</span>(len<span class="sc">~</span><span class="fu">interaction</span> (supp,dose),<span class="at">data=</span>ToothGrowth) <span class="co"># sensitivity non-normality +++</span></span>
<span id="cb43-7"><a href="#cb43-7" tabindex="-1"></a><span class="fu">leveneTest</span>(len<span class="sc">~</span><span class="fu">interaction</span> (supp,dose),<span class="at">data=</span>ToothGrowth) <span class="co"># sensitivity non-normality ++</span></span>
<span id="cb43-8"><a href="#cb43-8" tabindex="-1"></a><span class="fu">fligner.test</span>(len<span class="sc">~</span><span class="fu">interaction</span> (supp,dose),<span class="at">data=</span>ToothGrowth) <span class="co"># sensitivity non-normality +</span></span></code></pre></div>
<p class="alert">
<strong>Practice 7.4</strong> Create your <strong>own</strong>
<code>my.t.test</code> function (e.g. you will not use the above
<code>t.test</code> function but you will code your own function). You
will probably need to check at <code>?pt</code> in order to extract the
p-value. You can look at <code>getAnywhere("t.test.default")</code> for
the actual source code of the <code>t.test</code> function. Please use
your customized function to test the effect of <code>treatment</code> on
<em>length</em> at various <em>days</em> in the <code>rairuoho</code>
dataset. Interpret.
</p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
